{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "09453c0f",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import gzip"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "933d820c",
   "metadata": {},
   "outputs": [],
   "source": [
    "from collections import defaultdict\n",
    "mane_gff = '/common/zhangz2lab/shared/genomes/gff/MANE.GRCh38.v1.1.ensembl_genomic.gff.gz'\n",
    "\n",
    "def read_mane(fp):\n",
    "    cds_regions_loaded = defaultdict(list)\n",
    "    with gzip.open(fp, 'rt') as f:  # Open the GFF3 file\n",
    "        for line in f:\n",
    "            if not line.startswith('#'):  # Skip header lines\n",
    "                parts = line.strip().split('\\t')\n",
    "                if not parts[0] in ['chr%i'%i for i in range(1,23)] + ['chrX', 'chrY']:\n",
    "                    continue\n",
    "                if parts[2] == 'CDS':  # Focus on CDS entries\n",
    "                    attributes = parts[8]\n",
    "                    gene_id = [attr for attr in attributes.split(';') if 'gene_name' in attr][0].split('=')[1]\n",
    "                    strand = parts[6]\n",
    "                    start = parts[3]\n",
    "                    end = parts[4]\n",
    "                    reading_frame = parts[7]\n",
    "                    chrom = parts[0]\n",
    "                    cds_regions_loaded[gene_id].append((int(start), int(end), int(reading_frame), strand, chrom))\n",
    "                    \n",
    "    return cds_regions_loaded\n",
    "\n",
    "cds_regions_loaded = read_mane(mane_gff)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "44b7219b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[(3069260, 3069296, 0, '+', 'chr1'), (3186125, 3186474, 2, '+', 'chr1'), (3244087, 3244137, 0, '+', 'chr1'), (3385152, 3385286, 0, '+', 'chr1'), (3396491, 3396593, 0, '+', 'chr1'), (3402791, 3402998, 2, '+', 'chr1'), (3404739, 3404886, 1, '+', 'chr1'), (3405495, 3405648, 0, '+', 'chr1'), (3411384, 3412800, 2, '+', 'chr1'), (3414560, 3414647, 1, '+', 'chr1'), (3417828, 3417997, 0, '+', 'chr1'), (3418667, 3418744, 1, '+', 'chr1'), (3425581, 3425750, 1, '+', 'chr1'), (3426051, 3426225, 2, '+', 'chr1'), (3430872, 3431108, 1, '+', 'chr1'), (3431966, 3432140, 1, '+', 'chr1'), (3433677, 3433811, 0, '+', 'chr1')]\n"
     ]
    }
   ],
   "source": [
    "print(cds_regions_loaded.get(\"PRDM16\"))  # Example usage\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "b23dddca",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>cln_disease</th>\n",
       "      <th>cln_disease_id</th>\n",
       "      <th>cln_significance</th>\n",
       "      <th>cln_gene</th>\n",
       "      <th>cln_review</th>\n",
       "      <th>cln_hgvs</th>\n",
       "      <th>cln_disease_dbs</th>\n",
       "      <th>cln_clnvi</th>\n",
       "      <th>cln_af_exac</th>\n",
       "      <th>cln_af_esp</th>\n",
       "      <th>cln_af_tgp</th>\n",
       "      <th>cln_molecular_consequence</th>\n",
       "      <th>gene_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>925952</td>\n",
       "      <td>1019397</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>not_provided</td>\n",
       "      <td>MedGen:CN517202</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>SAMD11:148398</td>\n",
       "      <td>criteria_provided,_single_submitter</td>\n",
       "      <td>NC_000001.11:g.925952G&gt;A</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>SAMD11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>925956</td>\n",
       "      <td>1543320</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>not_provided</td>\n",
       "      <td>MedGen:CN517202</td>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>SAMD11:148398</td>\n",
       "      <td>criteria_provided,_single_submitter</td>\n",
       "      <td>NC_000001.11:g.925956C&gt;T</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>SO:0001819|synonymous_variant</td>\n",
       "      <td>SAMD11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>925969</td>\n",
       "      <td>1648427</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>not_provided</td>\n",
       "      <td>MedGen:CN517202</td>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>SAMD11:148398</td>\n",
       "      <td>criteria_provided,_single_submitter</td>\n",
       "      <td>NC_000001.11:g.925969C&gt;T</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>SAMD11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>925976</td>\n",
       "      <td>1362713</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>not_provided</td>\n",
       "      <td>MedGen:CN517202</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>SAMD11:148398</td>\n",
       "      <td>criteria_provided,_single_submitter</td>\n",
       "      <td>NC_000001.11:g.925976T&gt;C</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>SAMD11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>925986</td>\n",
       "      <td>1568423</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>not_provided</td>\n",
       "      <td>MedGen:CN517202</td>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>SAMD11:148398</td>\n",
       "      <td>criteria_provided,_single_submitter</td>\n",
       "      <td>NC_000001.11:g.925986C&gt;T</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>SO:0001819|synonymous_variant</td>\n",
       "      <td>SAMD11</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM     POS       ID REF ALT QUAL FILTER   cln_disease   cln_disease_id  \\\n",
       "0     1  925952  1019397   G   A    .      .  not_provided  MedGen:CN517202   \n",
       "1     1  925956  1543320   C   T    .      .  not_provided  MedGen:CN517202   \n",
       "2     1  925969  1648427   C   T    .      .  not_provided  MedGen:CN517202   \n",
       "3     1  925976  1362713   T   C    .      .  not_provided  MedGen:CN517202   \n",
       "4     1  925986  1568423   C   T    .      .  not_provided  MedGen:CN517202   \n",
       "\n",
       "         cln_significance       cln_gene                           cln_review  \\\n",
       "0  Uncertain_significance  SAMD11:148398  criteria_provided,_single_submitter   \n",
       "1           Likely_benign  SAMD11:148398  criteria_provided,_single_submitter   \n",
       "2           Likely_benign  SAMD11:148398  criteria_provided,_single_submitter   \n",
       "3  Uncertain_significance  SAMD11:148398  criteria_provided,_single_submitter   \n",
       "4           Likely_benign  SAMD11:148398  criteria_provided,_single_submitter   \n",
       "\n",
       "                   cln_hgvs cln_disease_dbs cln_clnvi cln_af_exac cln_af_esp  \\\n",
       "0  NC_000001.11:g.925952G>A            None      None        None       None   \n",
       "1  NC_000001.11:g.925956C>T            None      None        None       None   \n",
       "2  NC_000001.11:g.925969C>T            None      None        None       None   \n",
       "3  NC_000001.11:g.925976T>C            None      None        None       None   \n",
       "4  NC_000001.11:g.925986C>T            None      None        None       None   \n",
       "\n",
       "  cln_af_tgp      cln_molecular_consequence gene_name  \n",
       "0       None    SO:0001583|missense_variant    SAMD11  \n",
       "1       None  SO:0001819|synonymous_variant    SAMD11  \n",
       "2       None    SO:0001583|missense_variant    SAMD11  \n",
       "3       None    SO:0001583|missense_variant    SAMD11  \n",
       "4       None  SO:0001819|synonymous_variant    SAMD11  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def parse_string_to_dict(info_string):\n",
    "    \"\"\"Convert a VCF INFO field string to a dictionary.\"\"\"\n",
    "    info_dict = {}\n",
    "    for item in info_string.split(';'):\n",
    "        if '=' in item:\n",
    "            key, value = item.split('=', 1)\n",
    "            info_dict[key] = value\n",
    "    return info_dict\n",
    "\n",
    "def parse_clinvar_to_dataframe(filepath):\n",
    "    with gzip.open(filepath, 'rt') as file:\n",
    "        data = []  # List to store combined information for each variant\n",
    "        for line in file:\n",
    "            if line.startswith('#'):\n",
    "                continue  # Skip header lines\n",
    "            fields = line.strip().split('\\t')\n",
    "            info_field = fields[7]  # INFO field is the 8th field in VCF format\n",
    "            info_dict = parse_string_to_dict(info_field)\n",
    "            \n",
    "            # Combine first 7 fields with extracted INFO data\n",
    "            variant_info = fields[:7] + [\n",
    "                info_dict.get('CLNDN'),\n",
    "                info_dict.get('CLNDISDB'),\n",
    "                info_dict.get('CLNSIG'),\n",
    "                info_dict.get('GENEINFO'),\n",
    "                info_dict.get('CLNREVSTAT'),\n",
    "                info_dict.get('CLNHGVS'),\n",
    "                info_dict.get('CLNDISDBINCL'),\n",
    "                info_dict.get('CLNVI'),\n",
    "                info_dict.get('AF_EXAC'),\n",
    "                info_dict.get('AF_ESP'),\n",
    "                info_dict.get('AF_TGP'),\n",
    "                info_dict.get('MC'),\n",
    "            ]\n",
    "            \n",
    "            data.append(variant_info)\n",
    "        \n",
    "        # Convert the accumulated data into a pandas DataFrame\n",
    "        columns = ['CHROM', 'POS', 'ID', 'REF', 'ALT', 'QUAL', 'FILTER',\n",
    "                   'cln_disease', 'cln_disease_id', 'cln_significance', 'cln_gene',\n",
    "                   'cln_review', 'cln_hgvs', 'cln_disease_dbs', 'cln_clnvi',\n",
    "                   'cln_af_exac', 'cln_af_esp', 'cln_af_tgp', 'cln_molecular_consequence']\n",
    "        df = pd.DataFrame(data, columns=columns)\n",
    "        \n",
    "        return df\n",
    "\n",
    "# Example usage\n",
    "filepath = '/common/zhangz2lab/fzzhang/knowledge_graphs/wen-chyu/data/variant/clinvar/clinvar.vcf.gz' \n",
    "df = parse_clinvar_to_dataframe(filepath)\n",
    "\n",
    "df = df.dropna(subset=['cln_gene'])\n",
    "df['gene_name'] = df.apply(lambda row: row['cln_gene'].split(':')[0], axis=1)\n",
    "\n",
    "# Display the first few rows of the DataFrame\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "381c310b",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>cln_disease</th>\n",
       "      <th>cln_disease_id</th>\n",
       "      <th>cln_significance</th>\n",
       "      <th>cln_gene</th>\n",
       "      <th>cln_review</th>\n",
       "      <th>cln_hgvs</th>\n",
       "      <th>cln_disease_dbs</th>\n",
       "      <th>cln_clnvi</th>\n",
       "      <th>cln_af_exac</th>\n",
       "      <th>cln_af_esp</th>\n",
       "      <th>cln_af_tgp</th>\n",
       "      <th>cln_molecular_consequence</th>\n",
       "      <th>gene_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>5177</th>\n",
       "      <td>1</td>\n",
       "      <td>3186163</td>\n",
       "      <td>474439</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Primary_dilated_cardiomyopathy|Left_ventricula...</td>\n",
       "      <td>EFO:EFO_0000407,Human_Phenotype_Ontology:HP:00...</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>PRDM16:63976</td>\n",
       "      <td>criteria_provided,_multiple_submitters,_no_con...</td>\n",
       "      <td>NC_000001.11:g.3186163C&gt;T</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0.00004</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>PRDM16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5371</th>\n",
       "      <td>1</td>\n",
       "      <td>3402986</td>\n",
       "      <td>60728</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Primary_dilated_cardiomyopathy|Left_ventricula...</td>\n",
       "      <td>EFO:EFO_0000407,Human_Phenotype_Ontology:HP:00...</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>PRDM16:63976</td>\n",
       "      <td>criteria_provided,_multiple_submitters,_no_con...</td>\n",
       "      <td>NC_000001.11:g.3402986C&gt;T</td>\n",
       "      <td>None</td>\n",
       "      <td>UniProtKB_(protein):Q9HAZ2#VAR_070213|OMIM_All...</td>\n",
       "      <td>0.00003</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>PRDM16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5451</th>\n",
       "      <td>1</td>\n",
       "      <td>3405572</td>\n",
       "      <td>691817</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Left_ventricular_noncompaction_cardiomyopathy</td>\n",
       "      <td>Human_Phenotype_Ontology:HP:0011664,MedGen:C40...</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>PRDM16:63976</td>\n",
       "      <td>criteria_provided,_single_submitter</td>\n",
       "      <td>NC_000001.11:g.3405572C&gt;A</td>\n",
       "      <td>None</td>\n",
       "      <td>Klaassen_Lab,Charite_University_Medicine_Berli...</td>\n",
       "      <td>0.00001</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>PRDM16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5483</th>\n",
       "      <td>1</td>\n",
       "      <td>3411483</td>\n",
       "      <td>977171</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Primary_dilated_cardiomyopathy|Left_ventricula...</td>\n",
       "      <td>EFO:EFO_0000407,Human_Phenotype_Ontology:HP:00...</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>PRDM16:63976</td>\n",
       "      <td>criteria_provided,_single_submitter</td>\n",
       "      <td>NC_000001.11:g.3411483G&gt;A</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0.00001</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>PRDM16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5573</th>\n",
       "      <td>1</td>\n",
       "      <td>3411985</td>\n",
       "      <td>1526272</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Primary_dilated_cardiomyopathy</td>\n",
       "      <td>EFO:EFO_0000407,Human_Phenotype_Ontology:HP:00...</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>PRDM16:63976</td>\n",
       "      <td>no_assertion_criteria_provided</td>\n",
       "      <td>NC_000001.11:g.3411985C&gt;G</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>PRDM16</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     CHROM      POS       ID REF ALT QUAL FILTER  \\\n",
       "5177     1  3186163   474439   C   T    .      .   \n",
       "5371     1  3402986    60728   C   T    .      .   \n",
       "5451     1  3405572   691817   C   A    .      .   \n",
       "5483     1  3411483   977171   G   A    .      .   \n",
       "5573     1  3411985  1526272   C   G    .      .   \n",
       "\n",
       "                                            cln_disease  \\\n",
       "5177  Primary_dilated_cardiomyopathy|Left_ventricula...   \n",
       "5371  Primary_dilated_cardiomyopathy|Left_ventricula...   \n",
       "5451      Left_ventricular_noncompaction_cardiomyopathy   \n",
       "5483  Primary_dilated_cardiomyopathy|Left_ventricula...   \n",
       "5573                     Primary_dilated_cardiomyopathy   \n",
       "\n",
       "                                         cln_disease_id  \\\n",
       "5177  EFO:EFO_0000407,Human_Phenotype_Ontology:HP:00...   \n",
       "5371  EFO:EFO_0000407,Human_Phenotype_Ontology:HP:00...   \n",
       "5451  Human_Phenotype_Ontology:HP:0011664,MedGen:C40...   \n",
       "5483  EFO:EFO_0000407,Human_Phenotype_Ontology:HP:00...   \n",
       "5573  EFO:EFO_0000407,Human_Phenotype_Ontology:HP:00...   \n",
       "\n",
       "            cln_significance      cln_gene  \\\n",
       "5177  Uncertain_significance  PRDM16:63976   \n",
       "5371  Uncertain_significance  PRDM16:63976   \n",
       "5451  Uncertain_significance  PRDM16:63976   \n",
       "5483  Uncertain_significance  PRDM16:63976   \n",
       "5573  Uncertain_significance  PRDM16:63976   \n",
       "\n",
       "                                             cln_review  \\\n",
       "5177  criteria_provided,_multiple_submitters,_no_con...   \n",
       "5371  criteria_provided,_multiple_submitters,_no_con...   \n",
       "5451                criteria_provided,_single_submitter   \n",
       "5483                criteria_provided,_single_submitter   \n",
       "5573                     no_assertion_criteria_provided   \n",
       "\n",
       "                       cln_hgvs cln_disease_dbs  \\\n",
       "5177  NC_000001.11:g.3186163C>T            None   \n",
       "5371  NC_000001.11:g.3402986C>T            None   \n",
       "5451  NC_000001.11:g.3405572C>A            None   \n",
       "5483  NC_000001.11:g.3411483G>A            None   \n",
       "5573  NC_000001.11:g.3411985C>G            None   \n",
       "\n",
       "                                              cln_clnvi cln_af_exac  \\\n",
       "5177                                               None     0.00004   \n",
       "5371  UniProtKB_(protein):Q9HAZ2#VAR_070213|OMIM_All...     0.00003   \n",
       "5451  Klaassen_Lab,Charite_University_Medicine_Berli...     0.00001   \n",
       "5483                                               None     0.00001   \n",
       "5573                                               None        None   \n",
       "\n",
       "     cln_af_esp cln_af_tgp    cln_molecular_consequence gene_name  \n",
       "5177       None       None  SO:0001583|missense_variant    PRDM16  \n",
       "5371       None       None  SO:0001583|missense_variant    PRDM16  \n",
       "5451       None       None  SO:0001583|missense_variant    PRDM16  \n",
       "5483       None       None  SO:0001583|missense_variant    PRDM16  \n",
       "5573       None       None  SO:0001583|missense_variant    PRDM16  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "filtered_df = df[\n",
    "    (df['cln_disease'].str.contains('cardiomyopathy', case=False, na=False)) & \n",
    "     (df['cln_molecular_consequence'].str.contains('missense', case=False, na=False))\n",
    "]\n",
    "\n",
    "\n",
    "# Display the filtered DataFrame\n",
    "filtered_df.head()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "0640c788",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "    chrom      start        end         original_position  score\n",
      "0       X  120441764  120441765  chrX:119575620-119575620      1\n",
      "1       X  120442598  120442599  chrX:119576454-119576454      1\n",
      "2       X  101398032  101398033  chrX:100653021-100653021      1\n",
      "3       X  101398941  101398942  chrX:100653930-100653930      1\n",
      "4       X  101398894  101398895  chrX:100653883-100653883      1\n",
      "..    ...        ...        ...                       ...    ...\n",
      "435     3   38603947   38603948    chr3:38645439-38645439      1\n",
      "436     3   38581234   38581235    chr3:38622726-38622726      1\n",
      "437    11   47343050   47343051   chr11:47364602-47364602      1\n",
      "438     3   38580975   38580976    chr3:38622467-38622467      1\n",
      "439     3   38613771   38613772    chr3:38655263-38655263      1\n",
      "\n",
      "[440 rows x 5 columns]\n",
      "    CHROM        POS      ID REF ALT QUAL FILTER  \\\n",
      "0       1  156130708   48067   A   C    .      .   \n",
      "1       1  156130708  222692   A   G    .      .   \n",
      "2       1  156130741   14504   G   A    .      .   \n",
      "3       1  156135244   48090   G   A    .      .   \n",
      "4       1  156135940   48097   T   A    .      .   \n",
      "..    ...        ...     ...  ..  ..  ...    ...   \n",
      "412     X  120431414  180882   A   G    .      .   \n",
      "413     X  120441864  626562   T   C    .      .   \n",
      "414     X  120442599    9982   C   T    .      .   \n",
      "415     X  120447996   44432   T   A    .      .   \n",
      "416     X  120455477  179562   C   T    .      .   \n",
      "\n",
      "                                           cln_disease  \\\n",
      "0    Primary_dilated_cardiomyopathy|Charcot-Marie-T...   \n",
      "1    Primary_dilated_cardiomyopathy|Charcot-Marie-T...   \n",
      "2    Primary_dilated_cardiomyopathy|Cardiomyopathy|...   \n",
      "3    Primary_dilated_cardiomyopathy|Myocarditis|Cha...   \n",
      "4    Cardiomyopathy|Dilated_cardiomyopathy_1A|Charc...   \n",
      "..                                                 ...   \n",
      "412  Cardiomyopathy|Danon_disease|not_specified|Car...   \n",
      "413  Cardiomyopathy|Danon_disease|Cardiovascular_ph...   \n",
      "414  Hypertrophic_cardiomyopathy|Danon_disease|Card...   \n",
      "415  Cardiomyopathy|Hypertrophic_cardiomyopathy|Dan...   \n",
      "416  Cardiomyopathy|Danon_disease|not_specified|Car...   \n",
      "\n",
      "                                        cln_disease_id  \\\n",
      "0    EFO:EFO_0000407,Human_Phenotype_Ontology:HP:00...   \n",
      "1    EFO:EFO_0000407,Human_Phenotype_Ontology:HP:00...   \n",
      "2    EFO:EFO_0000407,Human_Phenotype_Ontology:HP:00...   \n",
      "3    EFO:EFO_0000407,Human_Phenotype_Ontology:HP:00...   \n",
      "4    Human_Phenotype_Ontology:HP:0001638,MONDO:MOND...   \n",
      "..                                                 ...   \n",
      "412  Human_Phenotype_Ontology:HP:0001638,MONDO:MOND...   \n",
      "413  Human_Phenotype_Ontology:HP:0001638,MONDO:MOND...   \n",
      "414  Human_Phenotype_Ontology:HP:0001639,MONDO:MOND...   \n",
      "415  Human_Phenotype_Ontology:HP:0001638,MONDO:MOND...   \n",
      "416  Human_Phenotype_Ontology:HP:0001638,MONDO:MOND...   \n",
      "\n",
      "                                 cln_significance  ... cln_af_exac cln_af_esp  \\\n",
      "0                    Pathogenic/Likely_pathogenic  ...        None       None   \n",
      "1    Conflicting_interpretations_of_pathogenicity  ...        None       None   \n",
      "2                    Pathogenic/Likely_pathogenic  ...        None       None   \n",
      "3    Conflicting_interpretations_of_pathogenicity  ...        None       None   \n",
      "4    Conflicting_interpretations_of_pathogenicity  ...     0.00006       None   \n",
      "..                                            ...  ...         ...        ...   \n",
      "412  Conflicting_interpretations_of_pathogenicity  ...     0.00017    0.00057   \n",
      "413  Conflicting_interpretations_of_pathogenicity  ...        None       None   \n",
      "414                                    Pathogenic  ...        None       None   \n",
      "415  Conflicting_interpretations_of_pathogenicity  ...        None    0.00047   \n",
      "416  Conflicting_interpretations_of_pathogenicity  ...     0.00006       None   \n",
      "\n",
      "    cln_af_tgp                          cln_molecular_consequence gene_name  \\\n",
      "0         None                        SO:0001583|missense_variant      LMNA   \n",
      "1         None                        SO:0001583|missense_variant      LMNA   \n",
      "2         None                        SO:0001583|missense_variant      LMNA   \n",
      "3         None                        SO:0001583|missense_variant      LMNA   \n",
      "4         None                        SO:0001583|missense_variant      LMNA   \n",
      "..         ...                                                ...       ...   \n",
      "412       None  SO:0001583|missense_variant,SO:0001627|intron_...     LAMP2   \n",
      "413       None                        SO:0001583|missense_variant     LAMP2   \n",
      "414       None                        SO:0001583|missense_variant     LAMP2   \n",
      "415    0.00026                        SO:0001583|missense_variant     LAMP2   \n",
      "416       None                        SO:0001583|missense_variant     LAMP2   \n",
      "\n",
      "    chrom      start        end         original_position score  \n",
      "0       1  156130707  156130708  chr1:156100499-156100499     1  \n",
      "1       1  156130707  156130708  chr1:156100499-156100499     1  \n",
      "2       1  156130740  156130741  chr1:156100532-156100532     1  \n",
      "3       1  156135243  156135244  chr1:156105035-156105035     1  \n",
      "4       1  156135939  156135940  chr1:156105731-156105731     1  \n",
      "..    ...        ...        ...                       ...   ...  \n",
      "412     X  120431413  120431414  chrX:119565269-119565269     1  \n",
      "413     X  120441863  120441864  chrX:119575719-119575719     1  \n",
      "414     X  120442598  120442599  chrX:119576454-119576454     1  \n",
      "415     X  120447995  120447996  chrX:119581851-119581851     1  \n",
      "416     X  120455476  120455477  chrX:119589332-119589332     1  \n",
      "\n",
      "[417 rows x 25 columns]\n",
      "0      A\n",
      "1      A\n",
      "2      G\n",
      "3      G\n",
      "4      T\n",
      "      ..\n",
      "412    A\n",
      "413    T\n",
      "414    C\n",
      "415    T\n",
      "416    C\n",
      "Name: REF, Length: 417, dtype: object\n",
      "  CHROM      POS       ID REF ALT QUAL FILTER  \\\n",
      "0     1  3186163   474439   C   T    .      .   \n",
      "1     1  3402986    60728   C   T    .      .   \n",
      "2     1  3405572   691817   C   A    .      .   \n",
      "3     1  3411483   977171   G   A    .      .   \n",
      "4     1  3411985  1526272   C   G    .      .   \n",
      "\n",
      "                                         cln_disease  \\\n",
      "0  Primary_dilated_cardiomyopathy|Left_ventricula...   \n",
      "1  Primary_dilated_cardiomyopathy|Left_ventricula...   \n",
      "2      Left_ventricular_noncompaction_cardiomyopathy   \n",
      "3  Primary_dilated_cardiomyopathy|Left_ventricula...   \n",
      "4                     Primary_dilated_cardiomyopathy   \n",
      "\n",
      "                                      cln_disease_id        cln_significance  \\\n",
      "0  EFO:EFO_0000407,Human_Phenotype_Ontology:HP:00...  Uncertain_significance   \n",
      "1  EFO:EFO_0000407,Human_Phenotype_Ontology:HP:00...  Uncertain_significance   \n",
      "2  Human_Phenotype_Ontology:HP:0011664,MedGen:C40...  Uncertain_significance   \n",
      "3  EFO:EFO_0000407,Human_Phenotype_Ontology:HP:00...  Uncertain_significance   \n",
      "4  EFO:EFO_0000407,Human_Phenotype_Ontology:HP:00...  Uncertain_significance   \n",
      "\n",
      "   ... cln_af_exac cln_af_esp cln_af_tgp    cln_molecular_consequence  \\\n",
      "0  ...     0.00004       None       None  SO:0001583|missense_variant   \n",
      "1  ...     0.00003       None       None  SO:0001583|missense_variant   \n",
      "2  ...     0.00001       None       None  SO:0001583|missense_variant   \n",
      "3  ...     0.00001       None       None  SO:0001583|missense_variant   \n",
      "4  ...        None       None       None  SO:0001583|missense_variant   \n",
      "\n",
      "  gene_name chrom start end original_position score  \n",
      "0    PRDM16   NaN   NaN NaN               NaN   NaN  \n",
      "1    PRDM16   NaN   NaN NaN               NaN   NaN  \n",
      "2    PRDM16   NaN   NaN NaN               NaN   NaN  \n",
      "3    PRDM16   NaN   NaN NaN               NaN   NaN  \n",
      "4    PRDM16   NaN   NaN NaN               NaN   NaN  \n",
      "\n",
      "[5 rows x 25 columns]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_729374/1905561337.py:6: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  filtered_df['CHROM'] = filtered_df['CHROM'].astype(str)\n",
      "/tmp/ipykernel_729374/1905561337.py:10: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  filtered_df['POS'] = filtered_df['POS'].astype(int)\n"
     ]
    }
   ],
   "source": [
    "new_bed_df = pd.read_csv('output_bed_cm.csv', index_col=0)\n",
    "# Remove 'chr' prefix from 'chrom' column in new_bed_df\n",
    "new_bed_df['chrom'] = new_bed_df['chrom'].str.replace('chr', '')\n",
    "print(new_bed_df)\n",
    "# Convert 'chrom' columns to strings in both DataFrames if they are not already\n",
    "filtered_df['CHROM'] = filtered_df['CHROM'].astype(str)\n",
    "new_bed_df['chrom'] = new_bed_df['chrom'].astype(str)\n",
    "\n",
    "# Ensure 'POS' in filtered_df and 'end' in new_bed_df are integers\n",
    "filtered_df['POS'] = filtered_df['POS'].astype(int)\n",
    "new_bed_df['end'] = new_bed_df['end'].astype(int)\n",
    "\n",
    "# Attempt the merge again\n",
    "matches = filtered_df.merge(new_bed_df, left_on=['CHROM', 'POS'], right_on=['chrom', 'end'])\n",
    "print(matches)\n",
    "print(matches['REF'])\n",
    "\n",
    "# Perform the merge to identify rows to keep (\"left_only\") and use suffixes to handle overlapping column names\n",
    "# We'll use suffixes that are unlikely to cause conflicts and are easy to filter out\n",
    "merged_df = pd.merge(filtered_df, matches, on=['CHROM', 'POS'], how='left', indicator=True, suffixes=('', '_drop'))\n",
    "\n",
    "# Filter to keep only rows from filtered_df that don't have matches in new_bed_df\n",
    "filtered_df_no_matches = merged_df[merged_df['_merge'] == 'left_only']\n",
    "\n",
    "# Now, remove columns that were added from 'matches' (those with '_drop' suffix) and the '_merge' column\n",
    "# This is done by filtering out columns that end with '_drop' and dropping the '_merge' column\n",
    "filtered_df_no_matches = filtered_df_no_matches.loc[:, ~filtered_df_no_matches.columns.str.endswith('_drop')].drop(columns=['_merge'])\n",
    "\n",
    "\n",
    "filtered_df = filtered_df_no_matches \n",
    "# Now, filtered_df_no_matches should have the original structure without \"_x\" or \"_y\" suffixes\n",
    "print(filtered_df_no_matches.head())\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "b9615ef5",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of 'benign' rows: 1150\n",
      "Number of 'pathogenic' rows: 530\n"
     ]
    }
   ],
   "source": [
    "# Filter the DataFrame to include only rows with 'cln_significance' containing \"benign\" or \"pathogenic\"\n",
    "filtered_df = filtered_df[filtered_df['cln_significance'].str.contains('benign|pathogenic', case=False)]\n",
    "\n",
    "# Filter out conflicting annotations\n",
    "filtered_df = filtered_df[~filtered_df['cln_significance'].str.contains('conflicting', case=False)]\n",
    "\n",
    "\n",
    "# Count rows where 'cln_significance' contains \"benign\"\n",
    "benign_count = filtered_df[filtered_df['cln_significance'].str.contains('benign', case=False)].shape[0]\n",
    "\n",
    "# Assuming you want to count rows where 'cln_significance' contains \"pathogenic\" (regardless of 'cln_molecular_consequence')\n",
    "pathogenic_count = filtered_df[filtered_df['cln_significance'].str.contains('pathogenic', case=False)].shape[0]\n",
    "\n",
    "print(f\"Number of 'benign' rows: {benign_count}\")\n",
    "print(f\"Number of 'pathogenic' rows: {pathogenic_count}\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "c526ff84",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Unique CHROM values in the DataFrame:\n",
      "1\n",
      "2\n",
      "3\n",
      "4\n",
      "5\n",
      "6\n",
      "7\n",
      "9\n",
      "10\n",
      "11\n",
      "12\n",
      "14\n",
      "15\n",
      "16\n",
      "17\n",
      "18\n",
      "19\n",
      "20\n",
      "21\n",
      "22\n",
      "X\n",
      "Total number of unique CHROM values: 21\n"
     ]
    }
   ],
   "source": [
    "# Get unique CHROM values from the DataFrame\n",
    "unique_chrom_values = filtered_df['CHROM'].unique()\n",
    "\n",
    "# Print all unique CHROM values\n",
    "print(\"Unique CHROM values in the DataFrame:\")\n",
    "for chrom in unique_chrom_values:\n",
    "    print(chrom)\n",
    "\n",
    "# Count the number of unique CHROM values\n",
    "number_of_unique_chrom_values = len(unique_chrom_values)\n",
    "print(f\"Total number of unique CHROM values: {number_of_unique_chrom_values}\")\n",
    "filtered_df = filtered_df[~filtered_df['CHROM'].isin(['MT', 'NW_009646201.1'])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "4466316a",
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio.Seq import Seq\n",
    "\n",
    "# Function to translate DNA sequence to protein sequence\n",
    "def translate_dna_to_protein(dna_sequence):\n",
    "    dna_seq = Seq(dna_sequence)\n",
    "    protein_seq = dna_seq.translate(to_stop=True)  # to_stop=True will stop at the first stop codon\n",
    "    return str(protein_seq)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "77981112",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Get proteins for each gene\n",
    "from pyfaidx import Fasta\n",
    "from Bio.Seq import Seq\n",
    "from tqdm import tqdm\n",
    "\n",
    "reference_genome = Fasta('/common/zhangz2lab/zhanh/Clinvar/hg38.fa')\n",
    "\n",
    "def reverse_complement(seq):\n",
    "    complement = {'A': 'T', 'T': 'A', 'G': 'C', 'C': 'G'}\n",
    "    return ''.join(complement.get(base, base) for base in reversed(seq))\n",
    "\n",
    "\n",
    "codon_table = {\n",
    "    'ATA':'I', 'ATC':'I', 'ATT':'I', 'ATG':'M',\n",
    "    'ACA':'T', 'ACC':'T', 'ACG':'T', 'ACT':'T',\n",
    "    'AAC':'N', 'AAT':'N', 'AAA':'K', 'AAG':'K',\n",
    "    'AGC':'S', 'AGT':'S', 'AGA':'R', 'AGG':'R',\n",
    "    'CTA':'L', 'CTC':'L', 'CTG':'L', 'CTT':'L',\n",
    "    'CCA':'P', 'CCC':'P', 'CCG':'P', 'CCT':'P',\n",
    "    'CAC':'H', 'CAT':'H', 'CAA':'Q', 'CAG':'Q',\n",
    "    'CGA':'R', 'CGC':'R', 'CGG':'R', 'CGT':'R',\n",
    "    'GTA':'V', 'GTC':'V', 'GTG':'V', 'GTT':'V',\n",
    "    'GCA':'A', 'GCC':'A', 'GCG':'A', 'GCT':'A',\n",
    "    'GAC':'D', 'GAT':'D', 'GAA':'E', 'GAG':'E',\n",
    "    'GGA':'G', 'GGC':'G', 'GGG':'G', 'GGT':'G',\n",
    "    'TCA':'S', 'TCC':'S', 'TCG':'S', 'TCT':'S',\n",
    "    'TTC':'F', 'TTT':'F', 'TTA':'L', 'TTG':'L',\n",
    "    'TAC':'Y', 'TAT':'Y', 'TAA':'_', 'TAG':'_',\n",
    "    'TGC':'C', 'TGT':'C', 'TGA':'_', 'TGG':'W',\n",
    "}\n",
    "\n",
    "def translate_dna(sequence):\n",
    "    protein = \"\"\n",
    "    for i in range(0, len(sequence), 3):\n",
    "        codon = sequence[i:i+3]\n",
    "        if len(codon) == 3:\n",
    "            protein += codon_table.get(codon, 'X')  # 'X' for unknown codons\n",
    "    return protein\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "dc34ebaa",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "('-',\n",
       " 'MKRTHLFIVGIYFLSSCRAEEGLNFPTYDGKDRVVSLSEKNFKQVLKKYDLLCLYYHEPVSSDKVTQKQFQLKEIVLELVAQVLEHKAIGFVMVDAKKEAKLAKKLGFDEEGSLYILKGDRTIEFDGEFAADVLVEFLLDLIEDPVEIISSKLEVQAFERIEDYIKLIGFFKSEDSEYYKAFEEAAEHFQPYIKFFATFDKGVAKKLSLKMNEVDFYEPFMDEPIAIPNKPYTEEELVEFVKEHQRPTLRRLRPEEMFETWEDDLNGIHIVAFAEKSDPDGYEFLEILKQVARDNTDNPDLSILWIDPDDFPLLVAYWEKTFKIDLFRPQIGVVNVTDADSVWMEIPDDDDLPTAEELEDWIEDVLSGXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX_',\n",
       " 'MKRTHLFIVGIYFLSSCRAEEGLNFPTYDGKDRVVSLSEKNFKQVLKKYDLLCLYYHEPVSSDKVTQKQFQLKEIVLELVAQVLEHKAIGFVMVDAKKEAKLAKKLGFDEEGSLYILKGDRTIEFDGEFAADVLVEFLLDLIEDPVEIISSKLEVQAFERIEDYIKLIGFFKSEDSEYYKAFEEAAEHFQPYIKFFATFDKGVAKKLSLKMNEVDFYEPFMDEPIAIPNKPYTEEELVEFVKEHQRPTLRRLRPEEMFETWEDDLNGIHIVAFAEKSDPDGYEFLEILKQVARDNTDNPDLSILWIDPDDFPLLVAYWEKTFKIDLFRPQIGVVNVTDADSVWMEIPDDDDLPTAEELEDWIEDVLSGKINTEDDDEDDDDDDNSDEEDNDDSDDDDDE')"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene = 'CASQ2'\n",
    "dna = ''\n",
    "\n",
    "for cds_start, cds_end, _, strand, chromosome in cds_regions_loaded[gene]:\n",
    "    if strand == '+':\n",
    "        dna += reference_genome[chromosome][cds_start-1:cds_end].seq\n",
    "    else:\n",
    "        dna += reference_genome[chromosome][cds_start-1:cds_end].reverse.complement.seq\n",
    "\n",
    "# why different? Biopython's translation matches UCSC genome browser\n",
    "strand, translate_dna(dna), translate_dna_to_protein(dna)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "c126dd66",
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "flank_amount = 500\n",
    "\n",
    "sequences = []\n",
    "\n",
    "for index, variant in filtered_df.iterrows():\n",
    "    gene_name = variant['gene_name']\n",
    "    chromosome = \"chr\" + str(variant['CHROM'])\n",
    "    position = int(variant['POS'])\n",
    "    ref_allele = variant['REF']\n",
    "    alt_allele = variant['ALT']\n",
    "    label = variant['cln_significance']\n",
    "\n",
    "    cds_list = cds_regions_loaded[gene_name]  # Get the list of CDS regions for the gene\n",
    "    cds_list_sorted = sorted(cds_list, key=lambda x: x[0])  # Ensure the CDS regions are sorted\n",
    "\n",
    "\n",
    "    dna_pos = 0\n",
    "    is_within_cds = False\n",
    "    # Find the index of the CDS region that contains the variant\n",
    "    # cds_regions_loaded is sorted ascending for + strand, and descending for - strand; no need to re-sort\n",
    "    for i, cds_interval in enumerate(cds_regions_loaded[gene_name]):\n",
    "        cds_start = cds_interval[0]; cds_end = cds_interval[1]; strand = cds_interval[-2]\n",
    "        #print(gene_name, dna_pos, position, cds_interval)\n",
    "        if cds_start <= position <= cds_end:\n",
    "            if strand == '+':\n",
    "                dna_pos += position - cds_start\n",
    "            else:\n",
    "                dna_pos += cds_end - position\n",
    "            is_within_cds = True\n",
    "            break\n",
    "        dna_pos += cds_end - cds_start + 1\n",
    "    \n",
    "    prot_pos = dna_pos//3\n",
    "\n",
    "    if not is_within_cds:\n",
    "        continue\n",
    "\n",
    "    dna = ''\n",
    "    for cds_start, cds_end, _, strand, chromosome in cds_regions_loaded[gene_name]:\n",
    "        if strand == '+':\n",
    "            dna += reference_genome[chromosome][cds_start-1:cds_end].seq\n",
    "        else:\n",
    "            dna += reference_genome[chromosome][cds_start-1:cds_end].reverse.complement.seq\n",
    "\n",
    "    ref_allele_len = len(ref_allele)\n",
    "    if strand == \"+\":\n",
    "        is_ref_match = int(dna[dna_pos:(dna_pos+ref_allele_len)] == ref_allele)\n",
    "        ref_dna = dna[:dna_pos] + ref_allele + dna[dna_pos+ref_allele_len:]\n",
    "        alt_dna = dna[:dna_pos] + alt_allele + dna[dna_pos+ref_allele_len:]\n",
    "    else:\n",
    "        ref_allele = reverse_complement(ref_allele)\n",
    "        alt_allele = reverse_complement(alt_allele)\n",
    "        is_ref_match = int(dna[dna_pos:(dna_pos+ref_allele_len)] == ref_allele)\n",
    "        ref_dna = dna[:dna_pos] + ref_allele + dna[dna_pos+ref_allele_len:]\n",
    "        alt_dna = dna[:dna_pos] + alt_allele + dna[dna_pos+ref_allele_len:]\n",
    "\n",
    "    ref_prot = translate_dna_to_protein(ref_dna)\n",
    "    alt_prot = translate_dna_to_protein(alt_dna)\n",
    "    ref_prot_ = ref_prot[max(0, prot_pos-flank_amount):min(len(ref_prot), prot_pos+flank_amount)]\n",
    "    alt_prot_ = alt_prot[max(0, prot_pos-flank_amount):min(len(alt_prot), prot_pos+flank_amount)]\n",
    "    ref_dna_ = ref_dna[max(0, dna_pos-flank_amount*3):min(len(ref_dna), dna_pos+flank_amount*3)]\n",
    "    alt_dna_ = alt_dna[max(0, dna_pos-flank_amount*3):min(len(alt_dna), dna_pos+flank_amount*3)]\n",
    "\n",
    "    sequences.append({\n",
    "        'ref_prot': ref_prot_,\n",
    "        'alt_prot': alt_prot_,\n",
    "        'ref_dna': ref_dna_,\n",
    "        'alt_dna': alt_dna_,\n",
    "        'strand': strand,\n",
    "        'label': label,\n",
    "        'gene': gene_name,\n",
    "        'ref': ref_allele,\n",
    "        'alt': alt_allele,\n",
    "        'POS': position,\n",
    "        'is_ref_match': is_ref_match,\n",
    "        'molecular_consequences': variant['cln_molecular_consequence']\n",
    "    })\n",
    "\n",
    "sequences_df = pd.DataFrame(sequences)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "f6c70674",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ref_prot</th>\n",
       "      <th>alt_prot</th>\n",
       "      <th>ref_dna</th>\n",
       "      <th>alt_dna</th>\n",
       "      <th>strand</th>\n",
       "      <th>label</th>\n",
       "      <th>gene</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>POS</th>\n",
       "      <th>is_ref_match</th>\n",
       "      <th>molecular_consequences</th>\n",
       "      <th>labels</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>IHSTVKPFICEVCHKSYTQFSNLCRHKRMHADCRTQIKCKDCGQMF...</td>\n",
       "      <td>IHSTVKPFICEVCHKSYTQFSNLCRHKRMHADCRTQIKCKDCGQMF...</td>\n",
       "      <td>TCCACAGCACGGTGAAGCCTTTCATATGTGAGGTCTGCCACAAGTC...</td>\n",
       "      <td>TCCACAGCACGGTGAAGCCTTTCATATGTGAGGTCTGCCACAAGTC...</td>\n",
       "      <td>+</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>PRDM16</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>3414616</td>\n",
       "      <td>1</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GNPALPLVSAVSNSSQGTTAAAGPEEKFESRLEDSCVEKLKTRSSD...</td>\n",
       "      <td>GNPALPLVSAVSNSSQGTTAAAGPEEKFESRLEDSCVEKLKTRSSD...</td>\n",
       "      <td>GGGAACCCAGCCCTGCCCCTGGTCTCCGCCGTCAGCAACAGCAGCC...</td>\n",
       "      <td>GGGAACCCAGCCCTGCCCCTGGTCTCCGCCGTCAGCAACAGCAGCC...</td>\n",
       "      <td>+</td>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>PRDM16</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>3426095</td>\n",
       "      <td>1</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GSDFEDVNTTTGTDLDTTTGTGSDLDSDVDSDPDKDKGKGKSAEGQ...</td>\n",
       "      <td>GSDFEDVNTTTGTDLDTTTGTGSDLDSDVDSDPDKDKGKGKSAEGQ...</td>\n",
       "      <td>GGCAGTGACTTTGAGGACGTCAACACCACCACGGGGACCGACCTGG...</td>\n",
       "      <td>GGCAGTGACTTTGAGGACGTCAACACCACCACGGGGACCGACCTGG...</td>\n",
       "      <td>+</td>\n",
       "      <td>Benign</td>\n",
       "      <td>PRDM16</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>3430888</td>\n",
       "      <td>1</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MEPGEVKDRILENISLSVKKLQSYFAACEDEIPAIRNHDKVLQRLC...</td>\n",
       "      <td>MEPGEVKDRILENISLSVKKLQSYFAACEDETPAIRNHDKVLQRLC...</td>\n",
       "      <td>ATGGAGCCGGGGGAGGTGAAGGACCGGATCCTGGAGAACATCTCGC...</td>\n",
       "      <td>ATGGAGCCGGGGGAGGTGAAGGACCGGATCCTGGAGAACATCTCGC...</td>\n",
       "      <td>+</td>\n",
       "      <td>Benign</td>\n",
       "      <td>PLEKHM2</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>15716271</td>\n",
       "      <td>1</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MEPGEVKDRILENISLSVKKLQSYFAACEDEIPAIRNHDKVLQRLC...</td>\n",
       "      <td>MEPGEVKDRILENISLSVKKLQSYFAACEDEIPAIRNHDKVLQRLC...</td>\n",
       "      <td>ATGGAGCCGGGGGAGGTGAAGGACCGGATCCTGGAGAACATCTCGC...</td>\n",
       "      <td>ATGGAGCCGGGGGAGGTGAAGGACCGGATCCTGGAGAACATCTCGC...</td>\n",
       "      <td>+</td>\n",
       "      <td>Benign</td>\n",
       "      <td>PLEKHM2</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>15718548</td>\n",
       "      <td>1</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1640</th>\n",
       "      <td>MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...</td>\n",
       "      <td>MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...</td>\n",
       "      <td>ATGCCTCTGCACGTGAAGTGGCCGTTCCCCGCGGTGCCGCCGCTCA...</td>\n",
       "      <td>ATGCCTCTGCACGTGAAGTGGCCGTTCCCCGCGGTGCCGCCGCTCA...</td>\n",
       "      <td>+</td>\n",
       "      <td>Pathogenic/Likely_pathogenic</td>\n",
       "      <td>TAFAZZIN</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>154413544</td>\n",
       "      <td>1</td>\n",
       "      <td>SO:0001583|missense_variant,SO:0001619|non-cod...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1641</th>\n",
       "      <td>MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...</td>\n",
       "      <td>MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...</td>\n",
       "      <td>ATGCCTCTGCACGTGAAGTGGCCGTTCCCCGCGGTGCCGCCGCTCA...</td>\n",
       "      <td>ATGCCTCTGCACGTGAAGTGGCCGTTCCCCGCGGTGCCGCCGCTCA...</td>\n",
       "      <td>+</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>TAFAZZIN</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>154413552</td>\n",
       "      <td>1</td>\n",
       "      <td>SO:0001583|missense_variant,SO:0001619|non-cod...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1642</th>\n",
       "      <td>MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...</td>\n",
       "      <td>MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...</td>\n",
       "      <td>ATGCCTCTGCACGTGAAGTGGCCGTTCCCCGCGGTGCCGCCGCTCA...</td>\n",
       "      <td>ATGCCTCTGCACGTGAAGTGGCCGTTCCCCGCGGTGCCGCCGCTCA...</td>\n",
       "      <td>+</td>\n",
       "      <td>Benign/Likely_benign</td>\n",
       "      <td>TAFAZZIN</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>154414113</td>\n",
       "      <td>0</td>\n",
       "      <td>SO:0001583|missense_variant,SO:0001627|intron_...</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1643</th>\n",
       "      <td>MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...</td>\n",
       "      <td>MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...</td>\n",
       "      <td>ATGCCTCTGCACGTGAAGTGGCCGTTCCCCGCGGTGCCGCCGCTCA...</td>\n",
       "      <td>ATGCCTCTGCACGTGAAGTGGCCGTTCCCCGCGGTGCCGCCGCTCA...</td>\n",
       "      <td>+</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>TAFAZZIN</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>154420212</td>\n",
       "      <td>1</td>\n",
       "      <td>SO:0001583|missense_variant,SO:0001619|non-cod...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1644</th>\n",
       "      <td>MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...</td>\n",
       "      <td>MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...</td>\n",
       "      <td>ATGCCTCTGCACGTGAAGTGGCCGTTCCCCGCGGTGCCGCCGCTCA...</td>\n",
       "      <td>ATGCCTCTGCACGTGAAGTGGCCGTTCCCCGCGGTGCCGCCGCTCA...</td>\n",
       "      <td>+</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>TAFAZZIN</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>154420676</td>\n",
       "      <td>1</td>\n",
       "      <td>SO:0001583|missense_variant,SO:0001619|non-cod...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1645 rows  13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                               ref_prot  \\\n",
       "0     IHSTVKPFICEVCHKSYTQFSNLCRHKRMHADCRTQIKCKDCGQMF...   \n",
       "1     GNPALPLVSAVSNSSQGTTAAAGPEEKFESRLEDSCVEKLKTRSSD...   \n",
       "2     GSDFEDVNTTTGTDLDTTTGTGSDLDSDVDSDPDKDKGKGKSAEGQ...   \n",
       "3     MEPGEVKDRILENISLSVKKLQSYFAACEDEIPAIRNHDKVLQRLC...   \n",
       "4     MEPGEVKDRILENISLSVKKLQSYFAACEDEIPAIRNHDKVLQRLC...   \n",
       "...                                                 ...   \n",
       "1640  MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...   \n",
       "1641  MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...   \n",
       "1642  MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...   \n",
       "1643  MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...   \n",
       "1644  MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...   \n",
       "\n",
       "                                               alt_prot  \\\n",
       "0     IHSTVKPFICEVCHKSYTQFSNLCRHKRMHADCRTQIKCKDCGQMF...   \n",
       "1     GNPALPLVSAVSNSSQGTTAAAGPEEKFESRLEDSCVEKLKTRSSD...   \n",
       "2     GSDFEDVNTTTGTDLDTTTGTGSDLDSDVDSDPDKDKGKGKSAEGQ...   \n",
       "3     MEPGEVKDRILENISLSVKKLQSYFAACEDETPAIRNHDKVLQRLC...   \n",
       "4     MEPGEVKDRILENISLSVKKLQSYFAACEDEIPAIRNHDKVLQRLC...   \n",
       "...                                                 ...   \n",
       "1640  MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...   \n",
       "1641  MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...   \n",
       "1642  MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...   \n",
       "1643  MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...   \n",
       "1644  MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...   \n",
       "\n",
       "                                                ref_dna  \\\n",
       "0     TCCACAGCACGGTGAAGCCTTTCATATGTGAGGTCTGCCACAAGTC...   \n",
       "1     GGGAACCCAGCCCTGCCCCTGGTCTCCGCCGTCAGCAACAGCAGCC...   \n",
       "2     GGCAGTGACTTTGAGGACGTCAACACCACCACGGGGACCGACCTGG...   \n",
       "3     ATGGAGCCGGGGGAGGTGAAGGACCGGATCCTGGAGAACATCTCGC...   \n",
       "4     ATGGAGCCGGGGGAGGTGAAGGACCGGATCCTGGAGAACATCTCGC...   \n",
       "...                                                 ...   \n",
       "1640  ATGCCTCTGCACGTGAAGTGGCCGTTCCCCGCGGTGCCGCCGCTCA...   \n",
       "1641  ATGCCTCTGCACGTGAAGTGGCCGTTCCCCGCGGTGCCGCCGCTCA...   \n",
       "1642  ATGCCTCTGCACGTGAAGTGGCCGTTCCCCGCGGTGCCGCCGCTCA...   \n",
       "1643  ATGCCTCTGCACGTGAAGTGGCCGTTCCCCGCGGTGCCGCCGCTCA...   \n",
       "1644  ATGCCTCTGCACGTGAAGTGGCCGTTCCCCGCGGTGCCGCCGCTCA...   \n",
       "\n",
       "                                                alt_dna strand  \\\n",
       "0     TCCACAGCACGGTGAAGCCTTTCATATGTGAGGTCTGCCACAAGTC...      +   \n",
       "1     GGGAACCCAGCCCTGCCCCTGGTCTCCGCCGTCAGCAACAGCAGCC...      +   \n",
       "2     GGCAGTGACTTTGAGGACGTCAACACCACCACGGGGACCGACCTGG...      +   \n",
       "3     ATGGAGCCGGGGGAGGTGAAGGACCGGATCCTGGAGAACATCTCGC...      +   \n",
       "4     ATGGAGCCGGGGGAGGTGAAGGACCGGATCCTGGAGAACATCTCGC...      +   \n",
       "...                                                 ...    ...   \n",
       "1640  ATGCCTCTGCACGTGAAGTGGCCGTTCCCCGCGGTGCCGCCGCTCA...      +   \n",
       "1641  ATGCCTCTGCACGTGAAGTGGCCGTTCCCCGCGGTGCCGCCGCTCA...      +   \n",
       "1642  ATGCCTCTGCACGTGAAGTGGCCGTTCCCCGCGGTGCCGCCGCTCA...      +   \n",
       "1643  ATGCCTCTGCACGTGAAGTGGCCGTTCCCCGCGGTGCCGCCGCTCA...      +   \n",
       "1644  ATGCCTCTGCACGTGAAGTGGCCGTTCCCCGCGGTGCCGCCGCTCA...      +   \n",
       "\n",
       "                             label      gene ref alt        POS  is_ref_match  \\\n",
       "0                       Pathogenic    PRDM16   T   C    3414616             1   \n",
       "1                    Likely_benign    PRDM16   G   A    3426095             1   \n",
       "2                           Benign    PRDM16   G   A    3430888             1   \n",
       "3                           Benign   PLEKHM2   T   C   15716271             1   \n",
       "4                           Benign   PLEKHM2   G   A   15718548             1   \n",
       "...                            ...       ...  ..  ..        ...           ...   \n",
       "1640  Pathogenic/Likely_pathogenic  TAFAZZIN   G   A  154413544             1   \n",
       "1641             Likely_pathogenic  TAFAZZIN   G   A  154413552             1   \n",
       "1642          Benign/Likely_benign  TAFAZZIN   T   C  154414113             0   \n",
       "1643             Likely_pathogenic  TAFAZZIN   G   T  154420212             1   \n",
       "1644                    Pathogenic  TAFAZZIN   G   C  154420676             1   \n",
       "\n",
       "                                 molecular_consequences  labels  \n",
       "0                           SO:0001583|missense_variant       1  \n",
       "1                           SO:0001583|missense_variant       0  \n",
       "2                           SO:0001583|missense_variant       0  \n",
       "3                           SO:0001583|missense_variant       0  \n",
       "4                           SO:0001583|missense_variant       0  \n",
       "...                                                 ...     ...  \n",
       "1640  SO:0001583|missense_variant,SO:0001619|non-cod...       1  \n",
       "1641  SO:0001583|missense_variant,SO:0001619|non-cod...       1  \n",
       "1642  SO:0001583|missense_variant,SO:0001627|intron_...       0  \n",
       "1643  SO:0001583|missense_variant,SO:0001619|non-cod...       1  \n",
       "1644  SO:0001583|missense_variant,SO:0001619|non-cod...       1  \n",
       "\n",
       "[1645 rows x 13 columns]"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sequences_df.strand.value_counts()\n",
    "sequences_df['gene']\n",
    "# Assuming sequences_df is your DataFrame and 'gene' is the column you want to save\n",
    "sequences_df['gene'].to_csv('clinvar_cm.csv', index=False)\n",
    "df = pd.read_csv(\"/common/zhangz2lab/zhanh/Propath-2/PLLR_CM.csv\")\n",
    "# For the DataFrame loaded from CSV\n",
    "\n",
    "# For the sequences_df DataFrame\n",
    "\n",
    "#df['gene'] = sequences_df['gene']\n",
    "#df['label'] = sequences_df['label']\n",
    "#df.to_csv('/common/zhangz2lab/zhanh/Propath-2/PLLR_CM.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "e459f177",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ref_prot</th>\n",
       "      <th>alt_prot</th>\n",
       "      <th>ref_dna</th>\n",
       "      <th>alt_dna</th>\n",
       "      <th>strand</th>\n",
       "      <th>label</th>\n",
       "      <th>gene</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>POS</th>\n",
       "      <th>is_ref_match</th>\n",
       "      <th>molecular_consequences</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>MNDISQKAEILLSSSKPVPKTYVPKLGKGDVKDKFEAMQRAREERN...</td>\n",
       "      <td>MNDISQKAEILLSSSKPVPKTYVPKLGKGDVKDKFEAMQRAREERN...</td>\n",
       "      <td>ATGAATGATATTTCCCAAAAGGCTGAGATTCTGCTTTCTTCATCTA...</td>\n",
       "      <td>ATGAATGATATTTCCCAAAAGGCTGAGATTCTGCTTTCTTCATCTA...</td>\n",
       "      <td>+</td>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>NEXN</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>77926854</td>\n",
       "      <td>0</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>MNDISQKAEILLSSSKPVPKTYVPKLGKGDVKDKFEAMQRAREERN...</td>\n",
       "      <td>MNDISQKAEILLSSSKPVPKTYVPKLGKGDVKDKFEAMQRAREERN...</td>\n",
       "      <td>ATGAATGATATTTCCCAAAAGGCTGAGATTCTGCTTTCTTCATCTA...</td>\n",
       "      <td>ATGAATGATATTTCCCAAAAGGCTGAGATTCTGCTTTCTTCATCTA...</td>\n",
       "      <td>+</td>\n",
       "      <td>Benign</td>\n",
       "      <td>NEXN</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>77929446</td>\n",
       "      <td>0</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>KGDVKDKFEAMQRAREERNQRRSRDEKQRRKEQYIREREWNRRKQE...</td>\n",
       "      <td>KGDVKDKFEAMQRAREERNQRRSRDEKQRRKEQYIREREWNRRKQE...</td>\n",
       "      <td>AAGGGTGATGTAAAGGATAAGTTTGAAGCCATGCAGAGAGCCAGGG...</td>\n",
       "      <td>AAGGGTGATGTAAAGGATAAGTTTGAAGCCATGCAGAGAGCCAGGG...</td>\n",
       "      <td>+</td>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>NEXN</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>77942131</td>\n",
       "      <td>0</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101</th>\n",
       "      <td>MSDIEEVVEEYEEEEQEEAAVEEEEDWREDEDEQEEAAEEDAEAEA...</td>\n",
       "      <td>MSDIEEVVEEYEEEEQEEAAVEEEEDWREDEDEQEEAAEEDAEAEA...</td>\n",
       "      <td>ATGTCTGACATAGAAGAGGTGGTGGAAGAGTACGAGGAGGAGGAGC...</td>\n",
       "      <td>ATGTCTGACATAGAAGAGGTGGTGGAAGAGTACGAGGAGGAGGAGC...</td>\n",
       "      <td>-</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>TNNT2</td>\n",
       "      <td>TC</td>\n",
       "      <td>GT</td>\n",
       "      <td>201365247</td>\n",
       "      <td>0</td>\n",
       "      <td>SO:0001583|missense_variant,SO:0001627|intron_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>168</th>\n",
       "      <td>TVSSDSVAKFAVKATGEPRPTAIWTKDGKAITQGGKYKLSEDKGGF...</td>\n",
       "      <td>TVSSDSVAKFAVKATGEPRPTAIWTKDGKAITQGGKYKLSEDKGGF...</td>\n",
       "      <td>TGTTTCTTCAGACAGTGTTGCTAAATTTGCAGTTAAGGCTACTGGA...</td>\n",
       "      <td>TGTTTCTTCAGACAGTGTTGCTAAATTTGCAGTTAAGGCTACTGGA...</td>\n",
       "      <td>-</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>TTN</td>\n",
       "      <td>AAGTAACATGG</td>\n",
       "      <td>TGAAAGAAAAA</td>\n",
       "      <td>178527198</td>\n",
       "      <td>0</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>174</th>\n",
       "      <td>RIMAEREDEELLRPVTTTQHLSEYKSELDFMSKEEKSRKKSRRQRE...</td>\n",
       "      <td>RIMAEREDEELLRPVTTTQHLSEYKSELDFMSKEEKSRKKSRRQRE...</td>\n",
       "      <td>GAATCATGGCTGAGAGGGAGGATGAAGAGTTGCTTCGCCCAGTTAC...</td>\n",
       "      <td>GAATCATGGCTGAGAGGGAGGATGAAGAGTTGCTTCGCCCAGTTAC...</td>\n",
       "      <td>-</td>\n",
       "      <td>Benign/Likely_benign</td>\n",
       "      <td>TTN</td>\n",
       "      <td>GC</td>\n",
       "      <td>TT</td>\n",
       "      <td>178530827</td>\n",
       "      <td>0</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>186</th>\n",
       "      <td>SDVSRDSVNLTWTEPASDGGSKITNYIVEKCATTAERWLRVGQARE...</td>\n",
       "      <td>SDVSRDSVNLTWTEPASDGGSKITNYIVEKCATTAERWLRVGQARE...</td>\n",
       "      <td>AGTGATGTCTCACGAGATTCTGTCAACTTAACATGGACTGAGCCAG...</td>\n",
       "      <td>AGTGATGTCTCACGAGATTCTGTCAACTTAACATGGACTGAGCCAG...</td>\n",
       "      <td>-</td>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>TTN</td>\n",
       "      <td>CA</td>\n",
       "      <td>TG</td>\n",
       "      <td>178534020</td>\n",
       "      <td>0</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>254</th>\n",
       "      <td>DHRYEFRVIARNAAGVFSEPSESTGAITARDEVDPPRISMDPKYKD...</td>\n",
       "      <td>DHRYEFRVIARNAAGVFSEPSESTGAITARDEVDPPRISMDPKYKD...</td>\n",
       "      <td>GATCACAGATATGAGTTCCGGGTTATAGCCCGAAATGCCGCAGGAG...</td>\n",
       "      <td>GATCACAGATATGAGTTCCGGGTTATAGCCCGAAATGCCGCAGGAG...</td>\n",
       "      <td>-</td>\n",
       "      <td>Benign/Likely_benign</td>\n",
       "      <td>TTN</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>178569412</td>\n",
       "      <td>0</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>557</th>\n",
       "      <td>MEHIQGAWKTISNGFGFKDAVFDGSSCISPTIVQQFGYQRRGSDDG...</td>\n",
       "      <td>MEHIQGAWKTISNGFGFKDAVFDGSSCISPTIVQQFGYQRRDSDDG...</td>\n",
       "      <td>ATGGAGCACATACAGGGAGCTTGGAAGACGATCAGCAATGGTTTTG...</td>\n",
       "      <td>ATGGAGCACATACAGGGAGCTTGGAAGACGATCAGCAATGGTTTTG...</td>\n",
       "      <td>-</td>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>RAF1</td>\n",
       "      <td>GC</td>\n",
       "      <td>AT</td>\n",
       "      <td>12618597</td>\n",
       "      <td>0</td>\n",
       "      <td>SO:0001583|missense_variant,SO:0001619|non-cod...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>599</th>\n",
       "      <td>MAAAAAAAAEQQSSNGPVKKSMREKAVERRSVNKEHNSNFKAGYIP...</td>\n",
       "      <td>VAAAAAAAAEQQSSNGPVKKSMREKAVERRSVNKEHNSNFKAGYIP...</td>\n",
       "      <td>AtggcggcagcggcggcggcggctgcAGAACAGCAAAGTTCCAATG...</td>\n",
       "      <td>GtggcggcagcggcggcggcggctgcAGAACAGCAAAGTTCCAATG...</td>\n",
       "      <td>-</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>SGCB</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>52038259</td>\n",
       "      <td>0</td>\n",
       "      <td>SO:0001582|initiatior_codon_variant,SO:0001583...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>706</th>\n",
       "      <td>MALSSAWRSVLPLWLLWSAACSRAASGDDNAFPFDIEGSSAVGRQD...</td>\n",
       "      <td>MALSSAWRSVLPLWLLWSAACSRAASGDDNAFPFDIEGSSAVGRQD...</td>\n",
       "      <td>ATGGCTTTGAGCTCAGCCTGGCGCTCGGTTCTGCCTCTGTGGCTCC...</td>\n",
       "      <td>ATGGCTTTGAGCTCAGCCTGGCGCTCGGTTCTGCCTCTGTGGCTCC...</td>\n",
       "      <td>-</td>\n",
       "      <td>Benign</td>\n",
       "      <td>LAMA4</td>\n",
       "      <td>AC</td>\n",
       "      <td>CA</td>\n",
       "      <td>112187567</td>\n",
       "      <td>0</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>735</th>\n",
       "      <td>MPLGLKPTCSVCKTTSSSMWKKGAQGEILCHHCTGRGGAGSGGAGS...</td>\n",
       "      <td>MPLGLKPTCSVCKTTSSSMWKKGAQGEILCHHCTGRGGAGSGGAGS...</td>\n",
       "      <td>ATGCCGCTGGGCCTGAAGCCCACCTGCAGCGTATGCAAGACCACGT...</td>\n",
       "      <td>ATGCCGCTGGGCCTGAAGCCCACCTGCAGCGTATGCAAGACCACGT...</td>\n",
       "      <td>+</td>\n",
       "      <td>Benign</td>\n",
       "      <td>GATAD1</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>92447889</td>\n",
       "      <td>0</td>\n",
       "      <td>SO:0001583|missense_variant,SO:0001619|non-cod...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>736</th>\n",
       "      <td>MPLGLKPTCSVCKTTSSSMWKKGAQGEILCHHCTGRGGAGSGGAGS...</td>\n",
       "      <td>MPLGLKPTCSVCKTTSSSMWKKGAQGEILCHHCTGRGGAGSGGAGS...</td>\n",
       "      <td>ATGCCGCTGGGCCTGAAGCCCACCTGCAGCGTATGCAAGACCACGT...</td>\n",
       "      <td>ATGCCGCTGGGCCTGAAGCCCACCTGCAGCGTATGCAAGACCACGT...</td>\n",
       "      <td>+</td>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>GATAD1</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>92447905</td>\n",
       "      <td>0</td>\n",
       "      <td>SO:0001583|missense_variant,SO:0001619|non-cod...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>798</th>\n",
       "      <td>MAALSGGGGGGAEPGQALFNGDMEPEAGAGAGAAASSAADPAIPEE...</td>\n",
       "      <td>MAALSGGGGGGAEPGQALFNGNMEPEAGAGAGAAASSAADPAIPEE...</td>\n",
       "      <td>ATGGcggcgctgagcggtggcggtggtggcggcgcggagccgggcc...</td>\n",
       "      <td>ATGGcggcgctgagcggtggcggtggtggcggcgcggagccgggcc...</td>\n",
       "      <td>-</td>\n",
       "      <td>Benign</td>\n",
       "      <td>BRAF</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>140924640</td>\n",
       "      <td>0</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>917</th>\n",
       "      <td>MVLAAAMSQDADPSGPEQPDRVACSVPGARASPAPSGPRGMQQPPP...</td>\n",
       "      <td>MVLAAAMSQDADPSGPEQPDRVACSVPGARASPAPSGPRGMQQQPP...</td>\n",
       "      <td>ATGGTGCTGGCAGCAGCCATGAGCCAGGACGCGGACCCCAGCGGTC...</td>\n",
       "      <td>ATGGTGCTGGCAGCAGCCATGAGCCAGGACGCGGACCCCAGCGGTC...</td>\n",
       "      <td>+</td>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>RBM20</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>110644585</td>\n",
       "      <td>0</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>995</th>\n",
       "      <td>MPNWGGGAKCGACEKTVYHAEEIQCNGRSFHKTCFHCMACRKALDS...</td>\n",
       "      <td>MPNWGGGAKCGACEKTVYHAEEIQCNGRSFHKTCFHCMACRKALDS...</td>\n",
       "      <td>ATGCCAAACTGGGGCGGAGGCGCAAAATGTGGAGCCTGTGAAAAGA...</td>\n",
       "      <td>ATGCCAAACTGGGGCGGAGGCGCAAAATGTGGAGCCTGTGAAAAGA...</td>\n",
       "      <td>-</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>CSRP3</td>\n",
       "      <td>TCGGA</td>\n",
       "      <td>AGGGG</td>\n",
       "      <td>19188253</td>\n",
       "      <td>0</td>\n",
       "      <td>SO:0001583|missense_variant,SO:0001627|intron_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1029</th>\n",
       "      <td>MPEPGKKPVSAFSKKPRSVEVAAGSPAVFEAETERAGVKVRWQRGG...</td>\n",
       "      <td>MPEPGKKPVSAFSKKPRSVEVAAGSPAVFEAETERAGVKVRWQRGG...</td>\n",
       "      <td>ATGCCTGAGCCGGGGAAGAAGCCAGTCTCAGCTTTTAGCAAGAAGC...</td>\n",
       "      <td>ATGCCTGAGCCGGGGAAGAAGCCAGTCTCAGCTTTTAGCAAGAAGC...</td>\n",
       "      <td>-</td>\n",
       "      <td>Benign/Likely_benign</td>\n",
       "      <td>MYBPC3</td>\n",
       "      <td>CA</td>\n",
       "      <td>TG</td>\n",
       "      <td>47348490</td>\n",
       "      <td>0</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1109</th>\n",
       "      <td>PLNETVVALYQKSSLKLMATLFSSYATADTGDSGKSKGGKKKGSSF...</td>\n",
       "      <td>PLNETVVALYQKSSLKLMATLFSSYATADTGDSGKSKGGKKKGSSF...</td>\n",
       "      <td>TCTCAACGAGACTGTTGTGGCCCTGTACCAGAAGTCCTCCCTCAAG...</td>\n",
       "      <td>TCTCAACGAGACTGTTGTGGCCCTGTACCAGAAGTCCTCCCTCAAG...</td>\n",
       "      <td>-</td>\n",
       "      <td>Benign/Likely_benign</td>\n",
       "      <td>MYH6</td>\n",
       "      <td>TG</td>\n",
       "      <td>CA</td>\n",
       "      <td>23392601</td>\n",
       "      <td>0</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1148</th>\n",
       "      <td>LFANYAGADAPIEKGKGKAKKGSSFQTVSALHRENLNKLMTNLRST...</td>\n",
       "      <td>LFANYAGADAPIEKGKGKAKKGSSFQTVSALHRENLNKLMTNLRST...</td>\n",
       "      <td>CTGTTTGCCAACTATGCTGGGGCTGATGCGCCTATTGAGAAGGGCA...</td>\n",
       "      <td>CTGTTTGCCAACTATGCTGGGGCTGATGCGCCTATTGAGAAGGGCA...</td>\n",
       "      <td>-</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>MYH7</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>23420213</td>\n",
       "      <td>0</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1414</th>\n",
       "      <td>MEAARPSGSWNTALCRLLLLTLAILIFASDACKNVTLHVPSKLDAE...</td>\n",
       "      <td>MEAARPSGSWNAALCRLLLLTLAILIFASDACKNVTLHVPSKLDAE...</td>\n",
       "      <td>ATGGAGGCAGCCCGCCCCTCCGGCTCCTGGAACACGGCCCTCTGCC...</td>\n",
       "      <td>ATGGAGGCAGCCCGCCCCTCCGGCTCCTGGAACGCAGCCCTCTGCC...</td>\n",
       "      <td>-</td>\n",
       "      <td>Benign/Likely_benign</td>\n",
       "      <td>DSC2</td>\n",
       "      <td>ACG</td>\n",
       "      <td>GCA</td>\n",
       "      <td>31101938</td>\n",
       "      <td>0</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1515</th>\n",
       "      <td>MSGGRFDFDDGGAYCGGWEGGKAHGHGLCTGPKGQGEYSGSWNFGF...</td>\n",
       "      <td>MSGGRFDFDDGGAYCGGWEGGKAHGHGLCTGPKGQGEYSGSWNFGF...</td>\n",
       "      <td>ATGAGTGGGGGCCGCTTCGACTTTGATGATGGAGGGGCGTACTGCG...</td>\n",
       "      <td>ATGAGTGGGGGCCGCTTCGACTTTGATGATGGAGGGGCGTACTGCG...</td>\n",
       "      <td>-</td>\n",
       "      <td>Benign</td>\n",
       "      <td>JPH2</td>\n",
       "      <td>GG</td>\n",
       "      <td>AA</td>\n",
       "      <td>44118607</td>\n",
       "      <td>0</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1528</th>\n",
       "      <td>MAAMAVALRGLGGRFRWRTQAVAGGVRGAARGAAAGQRDYDLLVVG...</td>\n",
       "      <td>MAAMAVALRGLGGRFRWRTQAVAGGVRGAARGAAAGQRDYDLLVVG...</td>\n",
       "      <td>ATGGCGGCAATGGCGGTGGCGCTGCGGGGATTAGGAGGGCGCTTCC...</td>\n",
       "      <td>ATGGCGGCAATGGCGGTGGCGCTGCGGGGATTAGGAGGGCGCTTCC...</td>\n",
       "      <td>-</td>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>TXNRD2</td>\n",
       "      <td>TA</td>\n",
       "      <td>CG</td>\n",
       "      <td>19880694</td>\n",
       "      <td>0</td>\n",
       "      <td>SO:0001583|missense_variant,SO:0001619|non-cod...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1537</th>\n",
       "      <td>MAAMAVALRGLGGRFRWRTQAVAGGVRGAARGAAAGQRDYDLLVVG...</td>\n",
       "      <td>MAAMAVALRGLGGRFRWRTQAVEGGVRGAARGAAAGQRDYDLLVVG...</td>\n",
       "      <td>ATGGCGGCAATGGCGGTGGCGCTGCGGGGATTAGGAGGGCGCTTCC...</td>\n",
       "      <td>ATGGCGGCAATGGCGGTGGCGCTGCGGGGATTAGGAGGGCGCTTCC...</td>\n",
       "      <td>-</td>\n",
       "      <td>Benign/Likely_benign</td>\n",
       "      <td>TXNRD2</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>19941736</td>\n",
       "      <td>0</td>\n",
       "      <td>SO:0001583|missense_variant,SO:0001619|non-cod...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1642</th>\n",
       "      <td>MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...</td>\n",
       "      <td>MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...</td>\n",
       "      <td>ATGCCTCTGCACGTGAAGTGGCCGTTCCCCGCGGTGCCGCCGCTCA...</td>\n",
       "      <td>ATGCCTCTGCACGTGAAGTGGCCGTTCCCCGCGGTGCCGCCGCTCA...</td>\n",
       "      <td>+</td>\n",
       "      <td>Benign/Likely_benign</td>\n",
       "      <td>TAFAZZIN</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>154414113</td>\n",
       "      <td>0</td>\n",
       "      <td>SO:0001583|missense_variant,SO:0001627|intron_...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                               ref_prot  \\\n",
       "28    MNDISQKAEILLSSSKPVPKTYVPKLGKGDVKDKFEAMQRAREERN...   \n",
       "30    MNDISQKAEILLSSSKPVPKTYVPKLGKGDVKDKFEAMQRAREERN...   \n",
       "32    KGDVKDKFEAMQRAREERNQRRSRDEKQRRKEQYIREREWNRRKQE...   \n",
       "101   MSDIEEVVEEYEEEEQEEAAVEEEEDWREDEDEQEEAAEEDAEAEA...   \n",
       "168   TVSSDSVAKFAVKATGEPRPTAIWTKDGKAITQGGKYKLSEDKGGF...   \n",
       "174   RIMAEREDEELLRPVTTTQHLSEYKSELDFMSKEEKSRKKSRRQRE...   \n",
       "186   SDVSRDSVNLTWTEPASDGGSKITNYIVEKCATTAERWLRVGQARE...   \n",
       "254   DHRYEFRVIARNAAGVFSEPSESTGAITARDEVDPPRISMDPKYKD...   \n",
       "557   MEHIQGAWKTISNGFGFKDAVFDGSSCISPTIVQQFGYQRRGSDDG...   \n",
       "599   MAAAAAAAAEQQSSNGPVKKSMREKAVERRSVNKEHNSNFKAGYIP...   \n",
       "706   MALSSAWRSVLPLWLLWSAACSRAASGDDNAFPFDIEGSSAVGRQD...   \n",
       "735   MPLGLKPTCSVCKTTSSSMWKKGAQGEILCHHCTGRGGAGSGGAGS...   \n",
       "736   MPLGLKPTCSVCKTTSSSMWKKGAQGEILCHHCTGRGGAGSGGAGS...   \n",
       "798   MAALSGGGGGGAEPGQALFNGDMEPEAGAGAGAAASSAADPAIPEE...   \n",
       "917   MVLAAAMSQDADPSGPEQPDRVACSVPGARASPAPSGPRGMQQPPP...   \n",
       "995   MPNWGGGAKCGACEKTVYHAEEIQCNGRSFHKTCFHCMACRKALDS...   \n",
       "1029  MPEPGKKPVSAFSKKPRSVEVAAGSPAVFEAETERAGVKVRWQRGG...   \n",
       "1109  PLNETVVALYQKSSLKLMATLFSSYATADTGDSGKSKGGKKKGSSF...   \n",
       "1148  LFANYAGADAPIEKGKGKAKKGSSFQTVSALHRENLNKLMTNLRST...   \n",
       "1414  MEAARPSGSWNTALCRLLLLTLAILIFASDACKNVTLHVPSKLDAE...   \n",
       "1515  MSGGRFDFDDGGAYCGGWEGGKAHGHGLCTGPKGQGEYSGSWNFGF...   \n",
       "1528  MAAMAVALRGLGGRFRWRTQAVAGGVRGAARGAAAGQRDYDLLVVG...   \n",
       "1537  MAAMAVALRGLGGRFRWRTQAVAGGVRGAARGAAAGQRDYDLLVVG...   \n",
       "1642  MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...   \n",
       "\n",
       "                                               alt_prot  \\\n",
       "28    MNDISQKAEILLSSSKPVPKTYVPKLGKGDVKDKFEAMQRAREERN...   \n",
       "30    MNDISQKAEILLSSSKPVPKTYVPKLGKGDVKDKFEAMQRAREERN...   \n",
       "32    KGDVKDKFEAMQRAREERNQRRSRDEKQRRKEQYIREREWNRRKQE...   \n",
       "101   MSDIEEVVEEYEEEEQEEAAVEEEEDWREDEDEQEEAAEEDAEAEA...   \n",
       "168   TVSSDSVAKFAVKATGEPRPTAIWTKDGKAITQGGKYKLSEDKGGF...   \n",
       "174   RIMAEREDEELLRPVTTTQHLSEYKSELDFMSKEEKSRKKSRRQRE...   \n",
       "186   SDVSRDSVNLTWTEPASDGGSKITNYIVEKCATTAERWLRVGQARE...   \n",
       "254   DHRYEFRVIARNAAGVFSEPSESTGAITARDEVDPPRISMDPKYKD...   \n",
       "557   MEHIQGAWKTISNGFGFKDAVFDGSSCISPTIVQQFGYQRRDSDDG...   \n",
       "599   VAAAAAAAAEQQSSNGPVKKSMREKAVERRSVNKEHNSNFKAGYIP...   \n",
       "706   MALSSAWRSVLPLWLLWSAACSRAASGDDNAFPFDIEGSSAVGRQD...   \n",
       "735   MPLGLKPTCSVCKTTSSSMWKKGAQGEILCHHCTGRGGAGSGGAGS...   \n",
       "736   MPLGLKPTCSVCKTTSSSMWKKGAQGEILCHHCTGRGGAGSGGAGS...   \n",
       "798   MAALSGGGGGGAEPGQALFNGNMEPEAGAGAGAAASSAADPAIPEE...   \n",
       "917   MVLAAAMSQDADPSGPEQPDRVACSVPGARASPAPSGPRGMQQQPP...   \n",
       "995   MPNWGGGAKCGACEKTVYHAEEIQCNGRSFHKTCFHCMACRKALDS...   \n",
       "1029  MPEPGKKPVSAFSKKPRSVEVAAGSPAVFEAETERAGVKVRWQRGG...   \n",
       "1109  PLNETVVALYQKSSLKLMATLFSSYATADTGDSGKSKGGKKKGSSF...   \n",
       "1148  LFANYAGADAPIEKGKGKAKKGSSFQTVSALHRENLNKLMTNLRST...   \n",
       "1414  MEAARPSGSWNAALCRLLLLTLAILIFASDACKNVTLHVPSKLDAE...   \n",
       "1515  MSGGRFDFDDGGAYCGGWEGGKAHGHGLCTGPKGQGEYSGSWNFGF...   \n",
       "1528  MAAMAVALRGLGGRFRWRTQAVAGGVRGAARGAAAGQRDYDLLVVG...   \n",
       "1537  MAAMAVALRGLGGRFRWRTQAVEGGVRGAARGAAAGQRDYDLLVVG...   \n",
       "1642  MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...   \n",
       "\n",
       "                                                ref_dna  \\\n",
       "28    ATGAATGATATTTCCCAAAAGGCTGAGATTCTGCTTTCTTCATCTA...   \n",
       "30    ATGAATGATATTTCCCAAAAGGCTGAGATTCTGCTTTCTTCATCTA...   \n",
       "32    AAGGGTGATGTAAAGGATAAGTTTGAAGCCATGCAGAGAGCCAGGG...   \n",
       "101   ATGTCTGACATAGAAGAGGTGGTGGAAGAGTACGAGGAGGAGGAGC...   \n",
       "168   TGTTTCTTCAGACAGTGTTGCTAAATTTGCAGTTAAGGCTACTGGA...   \n",
       "174   GAATCATGGCTGAGAGGGAGGATGAAGAGTTGCTTCGCCCAGTTAC...   \n",
       "186   AGTGATGTCTCACGAGATTCTGTCAACTTAACATGGACTGAGCCAG...   \n",
       "254   GATCACAGATATGAGTTCCGGGTTATAGCCCGAAATGCCGCAGGAG...   \n",
       "557   ATGGAGCACATACAGGGAGCTTGGAAGACGATCAGCAATGGTTTTG...   \n",
       "599   AtggcggcagcggcggcggcggctgcAGAACAGCAAAGTTCCAATG...   \n",
       "706   ATGGCTTTGAGCTCAGCCTGGCGCTCGGTTCTGCCTCTGTGGCTCC...   \n",
       "735   ATGCCGCTGGGCCTGAAGCCCACCTGCAGCGTATGCAAGACCACGT...   \n",
       "736   ATGCCGCTGGGCCTGAAGCCCACCTGCAGCGTATGCAAGACCACGT...   \n",
       "798   ATGGcggcgctgagcggtggcggtggtggcggcgcggagccgggcc...   \n",
       "917   ATGGTGCTGGCAGCAGCCATGAGCCAGGACGCGGACCCCAGCGGTC...   \n",
       "995   ATGCCAAACTGGGGCGGAGGCGCAAAATGTGGAGCCTGTGAAAAGA...   \n",
       "1029  ATGCCTGAGCCGGGGAAGAAGCCAGTCTCAGCTTTTAGCAAGAAGC...   \n",
       "1109  TCTCAACGAGACTGTTGTGGCCCTGTACCAGAAGTCCTCCCTCAAG...   \n",
       "1148  CTGTTTGCCAACTATGCTGGGGCTGATGCGCCTATTGAGAAGGGCA...   \n",
       "1414  ATGGAGGCAGCCCGCCCCTCCGGCTCCTGGAACACGGCCCTCTGCC...   \n",
       "1515  ATGAGTGGGGGCCGCTTCGACTTTGATGATGGAGGGGCGTACTGCG...   \n",
       "1528  ATGGCGGCAATGGCGGTGGCGCTGCGGGGATTAGGAGGGCGCTTCC...   \n",
       "1537  ATGGCGGCAATGGCGGTGGCGCTGCGGGGATTAGGAGGGCGCTTCC...   \n",
       "1642  ATGCCTCTGCACGTGAAGTGGCCGTTCCCCGCGGTGCCGCCGCTCA...   \n",
       "\n",
       "                                                alt_dna strand  \\\n",
       "28    ATGAATGATATTTCCCAAAAGGCTGAGATTCTGCTTTCTTCATCTA...      +   \n",
       "30    ATGAATGATATTTCCCAAAAGGCTGAGATTCTGCTTTCTTCATCTA...      +   \n",
       "32    AAGGGTGATGTAAAGGATAAGTTTGAAGCCATGCAGAGAGCCAGGG...      +   \n",
       "101   ATGTCTGACATAGAAGAGGTGGTGGAAGAGTACGAGGAGGAGGAGC...      -   \n",
       "168   TGTTTCTTCAGACAGTGTTGCTAAATTTGCAGTTAAGGCTACTGGA...      -   \n",
       "174   GAATCATGGCTGAGAGGGAGGATGAAGAGTTGCTTCGCCCAGTTAC...      -   \n",
       "186   AGTGATGTCTCACGAGATTCTGTCAACTTAACATGGACTGAGCCAG...      -   \n",
       "254   GATCACAGATATGAGTTCCGGGTTATAGCCCGAAATGCCGCAGGAG...      -   \n",
       "557   ATGGAGCACATACAGGGAGCTTGGAAGACGATCAGCAATGGTTTTG...      -   \n",
       "599   GtggcggcagcggcggcggcggctgcAGAACAGCAAAGTTCCAATG...      -   \n",
       "706   ATGGCTTTGAGCTCAGCCTGGCGCTCGGTTCTGCCTCTGTGGCTCC...      -   \n",
       "735   ATGCCGCTGGGCCTGAAGCCCACCTGCAGCGTATGCAAGACCACGT...      +   \n",
       "736   ATGCCGCTGGGCCTGAAGCCCACCTGCAGCGTATGCAAGACCACGT...      +   \n",
       "798   ATGGcggcgctgagcggtggcggtggtggcggcgcggagccgggcc...      -   \n",
       "917   ATGGTGCTGGCAGCAGCCATGAGCCAGGACGCGGACCCCAGCGGTC...      +   \n",
       "995   ATGCCAAACTGGGGCGGAGGCGCAAAATGTGGAGCCTGTGAAAAGA...      -   \n",
       "1029  ATGCCTGAGCCGGGGAAGAAGCCAGTCTCAGCTTTTAGCAAGAAGC...      -   \n",
       "1109  TCTCAACGAGACTGTTGTGGCCCTGTACCAGAAGTCCTCCCTCAAG...      -   \n",
       "1148  CTGTTTGCCAACTATGCTGGGGCTGATGCGCCTATTGAGAAGGGCA...      -   \n",
       "1414  ATGGAGGCAGCCCGCCCCTCCGGCTCCTGGAACGCAGCCCTCTGCC...      -   \n",
       "1515  ATGAGTGGGGGCCGCTTCGACTTTGATGATGGAGGGGCGTACTGCG...      -   \n",
       "1528  ATGGCGGCAATGGCGGTGGCGCTGCGGGGATTAGGAGGGCGCTTCC...      -   \n",
       "1537  ATGGCGGCAATGGCGGTGGCGCTGCGGGGATTAGGAGGGCGCTTCC...      -   \n",
       "1642  ATGCCTCTGCACGTGAAGTGGCCGTTCCCCGCGGTGCCGCCGCTCA...      +   \n",
       "\n",
       "                     label      gene          ref          alt        POS  \\\n",
       "28           Likely_benign      NEXN            G            A   77926854   \n",
       "30                  Benign      NEXN            A            C   77929446   \n",
       "32           Likely_benign      NEXN            G            C   77942131   \n",
       "101      Likely_pathogenic     TNNT2           TC           GT  201365247   \n",
       "168             Pathogenic       TTN  AAGTAACATGG  TGAAAGAAAAA  178527198   \n",
       "174   Benign/Likely_benign       TTN           GC           TT  178530827   \n",
       "186          Likely_benign       TTN           CA           TG  178534020   \n",
       "254   Benign/Likely_benign       TTN            T            C  178569412   \n",
       "557          Likely_benign      RAF1           GC           AT   12618597   \n",
       "599             Pathogenic      SGCB            A            G   52038259   \n",
       "706                 Benign     LAMA4           AC           CA  112187567   \n",
       "735                 Benign    GATAD1            G            A   92447889   \n",
       "736          Likely_benign    GATAD1            G            C   92447905   \n",
       "798                 Benign      BRAF            G            A  140924640   \n",
       "917          Likely_benign     RBM20            C            A  110644585   \n",
       "995             Pathogenic     CSRP3        TCGGA        AGGGG   19188253   \n",
       "1029  Benign/Likely_benign    MYBPC3           CA           TG   47348490   \n",
       "1109  Benign/Likely_benign      MYH6           TG           CA   23392601   \n",
       "1148     Likely_pathogenic      MYH7            G            A   23420213   \n",
       "1414  Benign/Likely_benign      DSC2          ACG          GCA   31101938   \n",
       "1515                Benign      JPH2           GG           AA   44118607   \n",
       "1528         Likely_benign    TXNRD2           TA           CG   19880694   \n",
       "1537  Benign/Likely_benign    TXNRD2            C            A   19941736   \n",
       "1642  Benign/Likely_benign  TAFAZZIN            T            C  154414113   \n",
       "\n",
       "      is_ref_match                             molecular_consequences  \n",
       "28               0                        SO:0001583|missense_variant  \n",
       "30               0                        SO:0001583|missense_variant  \n",
       "32               0                        SO:0001583|missense_variant  \n",
       "101              0  SO:0001583|missense_variant,SO:0001627|intron_...  \n",
       "168              0                        SO:0001583|missense_variant  \n",
       "174              0                        SO:0001583|missense_variant  \n",
       "186              0                        SO:0001583|missense_variant  \n",
       "254              0                        SO:0001583|missense_variant  \n",
       "557              0  SO:0001583|missense_variant,SO:0001619|non-cod...  \n",
       "599              0  SO:0001582|initiatior_codon_variant,SO:0001583...  \n",
       "706              0                        SO:0001583|missense_variant  \n",
       "735              0  SO:0001583|missense_variant,SO:0001619|non-cod...  \n",
       "736              0  SO:0001583|missense_variant,SO:0001619|non-cod...  \n",
       "798              0                        SO:0001583|missense_variant  \n",
       "917              0                        SO:0001583|missense_variant  \n",
       "995              0  SO:0001583|missense_variant,SO:0001627|intron_...  \n",
       "1029             0                        SO:0001583|missense_variant  \n",
       "1109             0                        SO:0001583|missense_variant  \n",
       "1148             0                        SO:0001583|missense_variant  \n",
       "1414             0                        SO:0001583|missense_variant  \n",
       "1515             0                        SO:0001583|missense_variant  \n",
       "1528             0  SO:0001583|missense_variant,SO:0001619|non-cod...  \n",
       "1537             0  SO:0001583|missense_variant,SO:0001619|non-cod...  \n",
       "1642             0  SO:0001583|missense_variant,SO:0001627|intron_...  "
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# no ref_allele mismatch\n",
    "sequences_df[sequences_df.is_ref_match==0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "11738124",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>label</th>\n",
       "      <th>molecular_consequences</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>463</th>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>SO:0001583|missense_variant,SO:0001627|intron_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>464</th>\n",
       "      <td>Benign/Likely_benign</td>\n",
       "      <td>SO:0001583|missense_variant,SO:0001627|intron_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>465</th>\n",
       "      <td>Benign/Likely_benign</td>\n",
       "      <td>SO:0001583|missense_variant,SO:0001627|intron_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>466</th>\n",
       "      <td>Benign/Likely_benign</td>\n",
       "      <td>SO:0001583|missense_variant,SO:0001627|intron_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>467</th>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>SO:0001583|missense_variant,SO:0001627|intron_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>723</th>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>825</th>\n",
       "      <td>Benign</td>\n",
       "      <td>SO:0001583|missense_variant,SO:0001819|synonym...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>826</th>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>SO:0001583|missense_variant,SO:0001819|synonym...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>827</th>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>SO:0001583|missense_variant,SO:0001819|synonym...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>828</th>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>SO:0001583|missense_variant,SO:0001819|synonym...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>896</th>\n",
       "      <td>Benign/Likely_benign</td>\n",
       "      <td>SO:0001583|missense_variant,SO:0001627|intron_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>897</th>\n",
       "      <td>Benign/Likely_benign</td>\n",
       "      <td>SO:0001583|missense_variant,SO:0001627|intron_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>898</th>\n",
       "      <td>Benign/Likely_benign</td>\n",
       "      <td>SO:0001583|missense_variant,SO:0001627|intron_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>903</th>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>SO:0001583|missense_variant,SO:0001627|intron_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>904</th>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>905</th>\n",
       "      <td>Benign</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>906</th>\n",
       "      <td>Benign/Likely_benign</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>907</th>\n",
       "      <td>Benign</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>908</th>\n",
       "      <td>Benign/Likely_benign</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>909</th>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>910</th>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>911</th>\n",
       "      <td>Benign</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>912</th>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>970</th>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>SO:0001583|missense_variant,SO:0001819|synonym...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>971</th>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>SO:0001583|missense_variant,SO:0001819|synonym...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>972</th>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>SO:0001583|missense_variant,SO:0001819|synonym...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>975</th>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>SO:0001583|missense_variant,SO:0001819|synonym...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>976</th>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>SO:0001583|missense_variant,SO:0001819|synonym...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>977</th>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>SO:0001583|missense_variant,SO:0001819|synonym...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>978</th>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>SO:0001583|missense_variant,SO:0001819|synonym...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>979</th>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>SO:0001583|missense_variant,SO:0001819|synonym...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>982</th>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>SO:0001583|missense_variant,SO:0001819|synonym...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>983</th>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>SO:0001583|missense_variant,SO:0001819|synonym...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>984</th>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>SO:0001583|missense_variant,SO:0001819|synonym...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>985</th>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>SO:0001583|missense_variant,SO:0001819|synonym...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>986</th>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>SO:0001583|missense_variant,SO:0001819|synonym...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>987</th>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>SO:0001583|missense_variant,SO:0001819|synonym...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>988</th>\n",
       "      <td>Benign</td>\n",
       "      <td>SO:0001583|missense_variant,SO:0001819|synonym...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>989</th>\n",
       "      <td>Benign</td>\n",
       "      <td>SO:0001583|missense_variant,SO:0001819|synonym...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>990</th>\n",
       "      <td>Benign/Likely_benign</td>\n",
       "      <td>SO:0001583|missense_variant,SO:0001819|synonym...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>991</th>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>SO:0001583|missense_variant,SO:0001819|synonym...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>992</th>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>SO:0001583|missense_variant,SO:0001819|synonym...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>993</th>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>SO:0001583|missense_variant,SO:0001819|synonym...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>994</th>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>SO:0001583|missense_variant,SO:0001819|synonym...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                    label                             molecular_consequences\n",
       "463         Likely_benign  SO:0001583|missense_variant,SO:0001627|intron_...\n",
       "464  Benign/Likely_benign  SO:0001583|missense_variant,SO:0001627|intron_...\n",
       "465  Benign/Likely_benign  SO:0001583|missense_variant,SO:0001627|intron_...\n",
       "466  Benign/Likely_benign  SO:0001583|missense_variant,SO:0001627|intron_...\n",
       "467         Likely_benign  SO:0001583|missense_variant,SO:0001627|intron_...\n",
       "723     Likely_pathogenic                        SO:0001583|missense_variant\n",
       "825                Benign  SO:0001583|missense_variant,SO:0001819|synonym...\n",
       "826         Likely_benign  SO:0001583|missense_variant,SO:0001819|synonym...\n",
       "827         Likely_benign  SO:0001583|missense_variant,SO:0001819|synonym...\n",
       "828         Likely_benign  SO:0001583|missense_variant,SO:0001819|synonym...\n",
       "896  Benign/Likely_benign  SO:0001583|missense_variant,SO:0001627|intron_...\n",
       "897  Benign/Likely_benign  SO:0001583|missense_variant,SO:0001627|intron_...\n",
       "898  Benign/Likely_benign  SO:0001583|missense_variant,SO:0001627|intron_...\n",
       "903            Pathogenic  SO:0001583|missense_variant,SO:0001627|intron_...\n",
       "904         Likely_benign                        SO:0001583|missense_variant\n",
       "905                Benign                        SO:0001583|missense_variant\n",
       "906  Benign/Likely_benign                        SO:0001583|missense_variant\n",
       "907                Benign                        SO:0001583|missense_variant\n",
       "908  Benign/Likely_benign                        SO:0001583|missense_variant\n",
       "909            Pathogenic                        SO:0001583|missense_variant\n",
       "910            Pathogenic                        SO:0001583|missense_variant\n",
       "911                Benign                        SO:0001583|missense_variant\n",
       "912            Pathogenic                        SO:0001583|missense_variant\n",
       "970         Likely_benign  SO:0001583|missense_variant,SO:0001819|synonym...\n",
       "971         Likely_benign  SO:0001583|missense_variant,SO:0001819|synonym...\n",
       "972         Likely_benign  SO:0001583|missense_variant,SO:0001819|synonym...\n",
       "975         Likely_benign  SO:0001583|missense_variant,SO:0001819|synonym...\n",
       "976         Likely_benign  SO:0001583|missense_variant,SO:0001819|synonym...\n",
       "977         Likely_benign  SO:0001583|missense_variant,SO:0001819|synonym...\n",
       "978         Likely_benign  SO:0001583|missense_variant,SO:0001819|synonym...\n",
       "979         Likely_benign  SO:0001583|missense_variant,SO:0001819|synonym...\n",
       "982         Likely_benign  SO:0001583|missense_variant,SO:0001819|synonym...\n",
       "983         Likely_benign  SO:0001583|missense_variant,SO:0001819|synonym...\n",
       "984     Likely_pathogenic  SO:0001583|missense_variant,SO:0001819|synonym...\n",
       "985         Likely_benign  SO:0001583|missense_variant,SO:0001819|synonym...\n",
       "986         Likely_benign  SO:0001583|missense_variant,SO:0001819|synonym...\n",
       "987         Likely_benign  SO:0001583|missense_variant,SO:0001819|synonym...\n",
       "988                Benign  SO:0001583|missense_variant,SO:0001819|synonym...\n",
       "989                Benign  SO:0001583|missense_variant,SO:0001819|synonym...\n",
       "990  Benign/Likely_benign  SO:0001583|missense_variant,SO:0001819|synonym...\n",
       "991         Likely_benign  SO:0001583|missense_variant,SO:0001819|synonym...\n",
       "992         Likely_benign  SO:0001583|missense_variant,SO:0001819|synonym...\n",
       "993         Likely_benign  SO:0001583|missense_variant,SO:0001819|synonym...\n",
       "994         Likely_benign  SO:0001583|missense_variant,SO:0001819|synonym..."
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# ref_prot != alt_prot unless other molecular_consequences exist\n",
    "sequences_df[sequences_df.ref_prot==sequences_df.alt_prot][['label','molecular_consequences']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "8edb608a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ref_prot</th>\n",
       "      <th>alt_prot</th>\n",
       "      <th>ref_dna</th>\n",
       "      <th>alt_dna</th>\n",
       "      <th>strand</th>\n",
       "      <th>label</th>\n",
       "      <th>gene</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>POS</th>\n",
       "      <th>is_ref_match</th>\n",
       "      <th>molecular_consequences</th>\n",
       "      <th>labels</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>IHSTVKPFICEVCHKSYTQFSNLCRHKRMHADCRTQIKCKDCGQMF...</td>\n",
       "      <td>IHSTVKPFICEVCHKSYTQFSNLCRHKRMHADCRTQIKCKDCGQMF...</td>\n",
       "      <td>TCCACAGCACGGTGAAGCCTTTCATATGTGAGGTCTGCCACAAGTC...</td>\n",
       "      <td>TCCACAGCACGGTGAAGCCTTTCATATGTGAGGTCTGCCACAAGTC...</td>\n",
       "      <td>+</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>PRDM16</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>3414616</td>\n",
       "      <td>1</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GNPALPLVSAVSNSSQGTTAAAGPEEKFESRLEDSCVEKLKTRSSD...</td>\n",
       "      <td>GNPALPLVSAVSNSSQGTTAAAGPEEKFESRLEDSCVEKLKTRSSD...</td>\n",
       "      <td>GGGAACCCAGCCCTGCCCCTGGTCTCCGCCGTCAGCAACAGCAGCC...</td>\n",
       "      <td>GGGAACCCAGCCCTGCCCCTGGTCTCCGCCGTCAGCAACAGCAGCC...</td>\n",
       "      <td>+</td>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>PRDM16</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>3426095</td>\n",
       "      <td>1</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GSDFEDVNTTTGTDLDTTTGTGSDLDSDVDSDPDKDKGKGKSAEGQ...</td>\n",
       "      <td>GSDFEDVNTTTGTDLDTTTGTGSDLDSDVDSDPDKDKGKGKSAEGQ...</td>\n",
       "      <td>GGCAGTGACTTTGAGGACGTCAACACCACCACGGGGACCGACCTGG...</td>\n",
       "      <td>GGCAGTGACTTTGAGGACGTCAACACCACCACGGGGACCGACCTGG...</td>\n",
       "      <td>+</td>\n",
       "      <td>Benign</td>\n",
       "      <td>PRDM16</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>3430888</td>\n",
       "      <td>1</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MEPGEVKDRILENISLSVKKLQSYFAACEDEIPAIRNHDKVLQRLC...</td>\n",
       "      <td>MEPGEVKDRILENISLSVKKLQSYFAACEDETPAIRNHDKVLQRLC...</td>\n",
       "      <td>ATGGAGCCGGGGGAGGTGAAGGACCGGATCCTGGAGAACATCTCGC...</td>\n",
       "      <td>ATGGAGCCGGGGGAGGTGAAGGACCGGATCCTGGAGAACATCTCGC...</td>\n",
       "      <td>+</td>\n",
       "      <td>Benign</td>\n",
       "      <td>PLEKHM2</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>15716271</td>\n",
       "      <td>1</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MEPGEVKDRILENISLSVKKLQSYFAACEDEIPAIRNHDKVLQRLC...</td>\n",
       "      <td>MEPGEVKDRILENISLSVKKLQSYFAACEDEIPAIRNHDKVLQRLC...</td>\n",
       "      <td>ATGGAGCCGGGGGAGGTGAAGGACCGGATCCTGGAGAACATCTCGC...</td>\n",
       "      <td>ATGGAGCCGGGGGAGGTGAAGGACCGGATCCTGGAGAACATCTCGC...</td>\n",
       "      <td>+</td>\n",
       "      <td>Benign</td>\n",
       "      <td>PLEKHM2</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>15718548</td>\n",
       "      <td>1</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1640</th>\n",
       "      <td>MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...</td>\n",
       "      <td>MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...</td>\n",
       "      <td>ATGCCTCTGCACGTGAAGTGGCCGTTCCCCGCGGTGCCGCCGCTCA...</td>\n",
       "      <td>ATGCCTCTGCACGTGAAGTGGCCGTTCCCCGCGGTGCCGCCGCTCA...</td>\n",
       "      <td>+</td>\n",
       "      <td>Pathogenic/Likely_pathogenic</td>\n",
       "      <td>TAFAZZIN</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>154413544</td>\n",
       "      <td>1</td>\n",
       "      <td>SO:0001583|missense_variant,SO:0001619|non-cod...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1641</th>\n",
       "      <td>MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...</td>\n",
       "      <td>MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...</td>\n",
       "      <td>ATGCCTCTGCACGTGAAGTGGCCGTTCCCCGCGGTGCCGCCGCTCA...</td>\n",
       "      <td>ATGCCTCTGCACGTGAAGTGGCCGTTCCCCGCGGTGCCGCCGCTCA...</td>\n",
       "      <td>+</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>TAFAZZIN</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>154413552</td>\n",
       "      <td>1</td>\n",
       "      <td>SO:0001583|missense_variant,SO:0001619|non-cod...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1642</th>\n",
       "      <td>MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...</td>\n",
       "      <td>MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...</td>\n",
       "      <td>ATGCCTCTGCACGTGAAGTGGCCGTTCCCCGCGGTGCCGCCGCTCA...</td>\n",
       "      <td>ATGCCTCTGCACGTGAAGTGGCCGTTCCCCGCGGTGCCGCCGCTCA...</td>\n",
       "      <td>+</td>\n",
       "      <td>Benign/Likely_benign</td>\n",
       "      <td>TAFAZZIN</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>154414113</td>\n",
       "      <td>0</td>\n",
       "      <td>SO:0001583|missense_variant,SO:0001627|intron_...</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1643</th>\n",
       "      <td>MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...</td>\n",
       "      <td>MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...</td>\n",
       "      <td>ATGCCTCTGCACGTGAAGTGGCCGTTCCCCGCGGTGCCGCCGCTCA...</td>\n",
       "      <td>ATGCCTCTGCACGTGAAGTGGCCGTTCCCCGCGGTGCCGCCGCTCA...</td>\n",
       "      <td>+</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>TAFAZZIN</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>154420212</td>\n",
       "      <td>1</td>\n",
       "      <td>SO:0001583|missense_variant,SO:0001619|non-cod...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1644</th>\n",
       "      <td>MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...</td>\n",
       "      <td>MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...</td>\n",
       "      <td>ATGCCTCTGCACGTGAAGTGGCCGTTCCCCGCGGTGCCGCCGCTCA...</td>\n",
       "      <td>ATGCCTCTGCACGTGAAGTGGCCGTTCCCCGCGGTGCCGCCGCTCA...</td>\n",
       "      <td>+</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>TAFAZZIN</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>154420676</td>\n",
       "      <td>1</td>\n",
       "      <td>SO:0001583|missense_variant,SO:0001619|non-cod...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1645 rows  13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                               ref_prot  \\\n",
       "0     IHSTVKPFICEVCHKSYTQFSNLCRHKRMHADCRTQIKCKDCGQMF...   \n",
       "1     GNPALPLVSAVSNSSQGTTAAAGPEEKFESRLEDSCVEKLKTRSSD...   \n",
       "2     GSDFEDVNTTTGTDLDTTTGTGSDLDSDVDSDPDKDKGKGKSAEGQ...   \n",
       "3     MEPGEVKDRILENISLSVKKLQSYFAACEDEIPAIRNHDKVLQRLC...   \n",
       "4     MEPGEVKDRILENISLSVKKLQSYFAACEDEIPAIRNHDKVLQRLC...   \n",
       "...                                                 ...   \n",
       "1640  MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...   \n",
       "1641  MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...   \n",
       "1642  MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...   \n",
       "1643  MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...   \n",
       "1644  MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...   \n",
       "\n",
       "                                               alt_prot  \\\n",
       "0     IHSTVKPFICEVCHKSYTQFSNLCRHKRMHADCRTQIKCKDCGQMF...   \n",
       "1     GNPALPLVSAVSNSSQGTTAAAGPEEKFESRLEDSCVEKLKTRSSD...   \n",
       "2     GSDFEDVNTTTGTDLDTTTGTGSDLDSDVDSDPDKDKGKGKSAEGQ...   \n",
       "3     MEPGEVKDRILENISLSVKKLQSYFAACEDETPAIRNHDKVLQRLC...   \n",
       "4     MEPGEVKDRILENISLSVKKLQSYFAACEDEIPAIRNHDKVLQRLC...   \n",
       "...                                                 ...   \n",
       "1640  MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...   \n",
       "1641  MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...   \n",
       "1642  MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...   \n",
       "1643  MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...   \n",
       "1644  MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...   \n",
       "\n",
       "                                                ref_dna  \\\n",
       "0     TCCACAGCACGGTGAAGCCTTTCATATGTGAGGTCTGCCACAAGTC...   \n",
       "1     GGGAACCCAGCCCTGCCCCTGGTCTCCGCCGTCAGCAACAGCAGCC...   \n",
       "2     GGCAGTGACTTTGAGGACGTCAACACCACCACGGGGACCGACCTGG...   \n",
       "3     ATGGAGCCGGGGGAGGTGAAGGACCGGATCCTGGAGAACATCTCGC...   \n",
       "4     ATGGAGCCGGGGGAGGTGAAGGACCGGATCCTGGAGAACATCTCGC...   \n",
       "...                                                 ...   \n",
       "1640  ATGCCTCTGCACGTGAAGTGGCCGTTCCCCGCGGTGCCGCCGCTCA...   \n",
       "1641  ATGCCTCTGCACGTGAAGTGGCCGTTCCCCGCGGTGCCGCCGCTCA...   \n",
       "1642  ATGCCTCTGCACGTGAAGTGGCCGTTCCCCGCGGTGCCGCCGCTCA...   \n",
       "1643  ATGCCTCTGCACGTGAAGTGGCCGTTCCCCGCGGTGCCGCCGCTCA...   \n",
       "1644  ATGCCTCTGCACGTGAAGTGGCCGTTCCCCGCGGTGCCGCCGCTCA...   \n",
       "\n",
       "                                                alt_dna strand  \\\n",
       "0     TCCACAGCACGGTGAAGCCTTTCATATGTGAGGTCTGCCACAAGTC...      +   \n",
       "1     GGGAACCCAGCCCTGCCCCTGGTCTCCGCCGTCAGCAACAGCAGCC...      +   \n",
       "2     GGCAGTGACTTTGAGGACGTCAACACCACCACGGGGACCGACCTGG...      +   \n",
       "3     ATGGAGCCGGGGGAGGTGAAGGACCGGATCCTGGAGAACATCTCGC...      +   \n",
       "4     ATGGAGCCGGGGGAGGTGAAGGACCGGATCCTGGAGAACATCTCGC...      +   \n",
       "...                                                 ...    ...   \n",
       "1640  ATGCCTCTGCACGTGAAGTGGCCGTTCCCCGCGGTGCCGCCGCTCA...      +   \n",
       "1641  ATGCCTCTGCACGTGAAGTGGCCGTTCCCCGCGGTGCCGCCGCTCA...      +   \n",
       "1642  ATGCCTCTGCACGTGAAGTGGCCGTTCCCCGCGGTGCCGCCGCTCA...      +   \n",
       "1643  ATGCCTCTGCACGTGAAGTGGCCGTTCCCCGCGGTGCCGCCGCTCA...      +   \n",
       "1644  ATGCCTCTGCACGTGAAGTGGCCGTTCCCCGCGGTGCCGCCGCTCA...      +   \n",
       "\n",
       "                             label      gene ref alt        POS  is_ref_match  \\\n",
       "0                       Pathogenic    PRDM16   T   C    3414616             1   \n",
       "1                    Likely_benign    PRDM16   G   A    3426095             1   \n",
       "2                           Benign    PRDM16   G   A    3430888             1   \n",
       "3                           Benign   PLEKHM2   T   C   15716271             1   \n",
       "4                           Benign   PLEKHM2   G   A   15718548             1   \n",
       "...                            ...       ...  ..  ..        ...           ...   \n",
       "1640  Pathogenic/Likely_pathogenic  TAFAZZIN   G   A  154413544             1   \n",
       "1641             Likely_pathogenic  TAFAZZIN   G   A  154413552             1   \n",
       "1642          Benign/Likely_benign  TAFAZZIN   T   C  154414113             0   \n",
       "1643             Likely_pathogenic  TAFAZZIN   G   T  154420212             1   \n",
       "1644                    Pathogenic  TAFAZZIN   G   C  154420676             1   \n",
       "\n",
       "                                 molecular_consequences  labels  \n",
       "0                           SO:0001583|missense_variant       1  \n",
       "1                           SO:0001583|missense_variant       0  \n",
       "2                           SO:0001583|missense_variant       0  \n",
       "3                           SO:0001583|missense_variant       0  \n",
       "4                           SO:0001583|missense_variant       0  \n",
       "...                                                 ...     ...  \n",
       "1640  SO:0001583|missense_variant,SO:0001619|non-cod...       1  \n",
       "1641  SO:0001583|missense_variant,SO:0001619|non-cod...       1  \n",
       "1642  SO:0001583|missense_variant,SO:0001627|intron_...       0  \n",
       "1643  SO:0001583|missense_variant,SO:0001619|non-cod...       1  \n",
       "1644  SO:0001583|missense_variant,SO:0001619|non-cod...       1  \n",
       "\n",
       "[1645 rows x 13 columns]"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# save\n",
    "sequences_df.to_csv('/common/zhangz2lab/zhanh/esm-variants/cropped/Clinvar_CM_protein.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "90195f68",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TCCACAGCACGGTGAAGCCTTTCATATGTGAGGTCTGCCACAAGTCCTACACGCAGTTCTCCAACCTGTGCCGGCACAAGCGGATGCACGCCGACTGCCGCACGCAGATCAAGTGCAAGGACTGTGGCCAGATGTTCAGCACTACCTCCTCCCTCAACAAGCACCGGCGCTTCTGCGAGGGCAAGAACCATTACACGCCGGGCGGCATCTTTGCCCCGGGCCTGCCCTTGACCCCCAGCCCCATGATGGACAAGGCAAAACCCTCCCCCAGCCTCAATCACGCCAGCCTGGGCTTCAACGAGTACTTTCCCTCCAGGCCGCACCCGGGGAGCCTGCCCTTCTCCACGGCGCCTCCCACGTTCCCCGCACTCACCCCCGGCTTCCCGGGCATCTTCCCTCCATCCTTGTACCCCCGGCCGCCTCTGCTACCTCCCACATCGCTGCTCAAGAGCCCCCTGAACCACACCCAGGACGCCAAGCTCCCCAGTCCCCTGGGGAACCCAGCCCTGCCCCTGGTCTCCGCCGTCAGCAACAGCAGCCAGGGCACGACGGCAGCTGCGGGGCCCGAGGAGAAGTTCGAGAGCCGCCTGGAGGACTCCTGTGTGGAGAAGCTGAAGACCAGGAGCAGCGACATGTCGGACGGCAGTGACTTTGAGGACGTCAACACCACCACGGGGACCGACCTGGACACGACCacggggacgggctcggacctggacagcgacgtggacagcgaccctgacaaggacaagggcaagggcaAGTCCGCCGAGGGCCAGCCCAAGTTTGGGGGCGGCTTGGCGCCCCCGGGGGCCCCGAACAGCGTGGCCGAGGTGCCTGTCTTCTATTCCCAGCACTCATTCTTCCCGCCACCCGACGAGCAGCTGCTGACTGCAACGGGCGCCGCCGGGGACTCCATCAAGGCCATCGCATCCATTGCCGAGAAGTACTTTGGCCCCGGCTTCATGGGGATGCAGGAGAAGAAGCTGGGCTCGCTCCCCTACCACTCGGCGTTCCCCTTCCAGTTCCTGCCCAACTTCCCCCACTCCCTTTACCCCTTCACGGACCGAGCCCTCGCCCACAACTTGCTGGTCAAGGCCGAGCCAAAGTCACCCCGGGACGCCCTCAAGGTGGGCGGCCCCAGTGCCGAGTGCCCCTTTGATCTCACCACCAAGCCCAAAGACGTGAAGCCCATCCTGCCCATGCCCAAGGGCCCCTCGGCCCCCGCATCCGGCGAGGAGCAGCCGCTGGACCTGAGCATCGGCAGCCGGGCCCGTGCCAGCCAAAACGGCGGCGGGCGGGAGCCCCGCAAGAACCACGTCTATGGGGAACGCAAGCTGGGCGCCGGCGAGGGGCTGCCCCAGGTGTGCCCGGCGCGGATGCCCCAGCAGCCCCCGCTCCACTACGCCAAGCCCTCGCCCTTCTTCATGGACCCCATCTACAGCAGGGTAGAAAAGCGGAAGGTCACAGACCCCGTGGGAGCCCTGAAGGAGAAGTACCTGCGGCCGTCCCCGCTGCTCTTCCACCCCCAGATGTCAGCCATAGAGACCATGACAGAGAAGCTGGAGAGCTTTGCAGCCATGAAGGCGGACTCGGGCAGCTCCCTGCAGCCCCTCCCCCACCACCCCTTCAACTTCCGGTCCCCACCCCCAACGCTCTCCGACCCCATCCTCAGGAAGGGCAAGGAGCGATACACGTGCAGGTACTGTGGGAAGATCTTCCCCAGATCAGCCAATCTCACCAGACACCTGAGGACGCACACTGGGGAGCAGCCGTACAGGTGTAAGTACTGCGACCGCTCCTTCAGCATCTCTTCGAACCTCCAGCGGCACGTCCGGAACATCCACAACAAGGAGAAGCCTTTCAAGTGCCACCTGTGCAACCGCTGCTTCGGGCAGCAGACCAACCTGGACCGGCACCTCAAGAAGCACGAGCACGAGAACGCACCAGTGAGCCAGCACCCCGGGGTCCTCACGAACCACCTGGGGACCAGCGCGTCCTCTCCCACCTCAGAGTCGGACAACCACGCACTTTTAGACGAGAAAGAAGACTCTTATTTCTCGGAAATCAGAAACTTTATTGCCAATAGTGAGATGAACCAAGCATCAACGCGAACAGAGAAACGGGCGGACATGCAGATCGTGGACGGCAGTGCCCAGTGTCCAGGCCTAGCCAGTgagaagcaggaggacgtggaggaggaggacgacgatgacctggaggaggacgatgaggaCAGCCTGGCCGGGAAGTCGCAGGATGACACCGTGTCCCCCGCACCCGAGCCCCAGGCCGCCTACGAGGATGAGGAGGATGAGGAGCCAGCCGCCTCCCTGGCCGTGGGCTTTGACCACACCCGAAGGTGTGCTGAGGACCACGAAGGCGGTCTGTTAGCTTTGGAGCCGATGCCGACTTTTGGGAAGGGGCTGGACCTCCGCAGAGCAGCTGAGGAAGCATTTGAAGTTAAAGATGTGCTTAATTCCACCTTAGATTCTGAGGCTTTAAAACATACACTGTGCAGGCAGGCTAAGAACCAGGCATATGCAATGATGCTGTCCCTTTCCGAAGACACTCCTCTCCACACCCCCTCCCAGGGTTCTCTGGACGCTTGGTTGAAGGTCACTGGAGCCACGTCGGAGTCTGGAGCATTTCACCCCATCAACCACCTCTGA\n",
      "TCCACAGCACGGTGAAGCCTTTCATATGTGAGGTCTGCCACAAGTCCTACACGCAGTTCTCCAACCTGTGCCGGCACAAGCGGATGCACGCCGACTGCCGCACGCAGATCAAGTGCAAGGACTGTGGCCAGATGTTCAGCACTACCTCCTCCCTCAACAAGCACCGGCGCTTCTGCGAGGGCAAGAACCATTACACGCCGGGCGGCATCTTTGCCCCGGGCCTGCCCTTGACCCCCAGCCCCATGATGGACAAGGCAAAACCCTCCCCCAGCCTCAATCACGCCAGCCTGGGCTTCAACGAGTACTTTCCCTCCAGGCCGCACCCGGGGAGCCTGCCCTTCTCCACGGCGCCTCCCACGTTCCCCGCACTCACCCCCGGCTTCCCGGGCATCTTCCCTCCATCCTTGTACCCCCGGCCGCCTCTGCTACCTCCCACATCGCTGCTCAAGAGCCCCCTGAACCACACCCAGGACGCCAAGCTCCCCAGTCCCCTGGGGAACCCAGCCCTGCCCCTGGTCTCCGCCGTCAGCAACAGCAGCCAGGGCACGACGGCAGCTGCGGGGCCCGAGGAGAAGTTCGAGAGCCGCCTGGAGGACTCCTGTGTGGAGAAGCTGAAGACCAGGAGCAGCGACATGTCGGACGGCAGTGACTTTGAGGACGTCAACACCACCACGGGGACCGACCTGGACACGACCacggggacgggctcggacctggacagcgacgtggacagcgaccctgacaaggacaagggcaagggcaAGTCCGCCGAGGGCCAGCCCAAGTTTGGGGGCGGCTTGGCGCCCCCGGGGGCCCCGAACAGCGTGGCCGAGGTGCCTGTCTTCTATTCCCAGCACTCATTCTTCCCGCCACCCGACGAGCAGCTGCTGACTGCAACGGGCGCCGCCGGGGACTCCATCAAGGCCATCGCATCCATTGCCGAGAAGTACTTTGGCCCCGGCTTCATGGGGATGCAGGAGAAGAAGCTGGGCTCGCTCCCCTACCACTCGGCGTTCCCCTTCCAGTTCCTGCCCAACTTCCCCCACTCCCTTTACCCCTTCACGGACCGAGCCCTCGCCCACAACTTGCTGGTCAAGGCCGAGCCAAAGTCACCCCGGGACGCCCTCAAGGTGGGCGGCCCCAGTGCCGAGTGCCCCTTTGATCTCACCACCAAGCCCAAAGACGTGAAGCCCATCCTGCCCATGCCCAAGGGCCCCTCGGCCCCCGCATCCGGCGAGGAGCAGCCGCTGGACCTGAGCATCGGCAGCCGGGCCCGTGCCAGCCAAAACGGCGGCGGGCGGGAGCCCCGCAAGAACCACGTCTATGGGGAACGCAAGCTGGGCGCCGGCGAGGGGCTGCCCCAGGTGTGCCCGGCGCGGATGCCCCAGCAGCCCCCGCTCCACTACGCCAAGCCCTCGCCCTTCTTCATGGACCCCATCTACAGCAGGGTAGAAAAGCGGAAGGTCACAGACCCCGTGGGAGCCCTGAAGGAGAAGTACCCGCGGCCGTCCCCGCTGCTCTTCCACCCCCAGATGTCAGCCATAGAGACCATGACAGAGAAGCTGGAGAGCTTTGCAGCCATGAAGGCGGACTCGGGCAGCTCCCTGCAGCCCCTCCCCCACCACCCCTTCAACTTCCGGTCCCCACCCCCAACGCTCTCCGACCCCATCCTCAGGAAGGGCAAGGAGCGATACACGTGCAGGTACTGTGGGAAGATCTTCCCCAGATCAGCCAATCTCACCAGACACCTGAGGACGCACACTGGGGAGCAGCCGTACAGGTGTAAGTACTGCGACCGCTCCTTCAGCATCTCTTCGAACCTCCAGCGGCACGTCCGGAACATCCACAACAAGGAGAAGCCTTTCAAGTGCCACCTGTGCAACCGCTGCTTCGGGCAGCAGACCAACCTGGACCGGCACCTCAAGAAGCACGAGCACGAGAACGCACCAGTGAGCCAGCACCCCGGGGTCCTCACGAACCACCTGGGGACCAGCGCGTCCTCTCCCACCTCAGAGTCGGACAACCACGCACTTTTAGACGAGAAAGAAGACTCTTATTTCTCGGAAATCAGAAACTTTATTGCCAATAGTGAGATGAACCAAGCATCAACGCGAACAGAGAAACGGGCGGACATGCAGATCGTGGACGGCAGTGCCCAGTGTCCAGGCCTAGCCAGTgagaagcaggaggacgtggaggaggaggacgacgatgacctggaggaggacgatgaggaCAGCCTGGCCGGGAAGTCGCAGGATGACACCGTGTCCCCCGCACCCGAGCCCCAGGCCGCCTACGAGGATGAGGAGGATGAGGAGCCAGCCGCCTCCCTGGCCGTGGGCTTTGACCACACCCGAAGGTGTGCTGAGGACCACGAAGGCGGTCTGTTAGCTTTGGAGCCGATGCCGACTTTTGGGAAGGGGCTGGACCTCCGCAGAGCAGCTGAGGAAGCATTTGAAGTTAAAGATGTGCTTAATTCCACCTTAGATTCTGAGGCTTTAAAACATACACTGTGCAGGCAGGCTAAGAACCAGGCATATGCAATGATGCTGTCCCTTTCCGAAGACACTCCTCTCCACACCCCCTCCCAGGGTTCTCTGGACGCTTGGTTGAAGGTCACTGGAGCCACGTCGGAGTCTGGAGCATTTCACCCCATCAACCACCTCTGA\n"
     ]
    }
   ],
   "source": [
    "print(sequences_df['ref_dna'].iloc[0])\n",
    "print(sequences_df['alt_dna'].iloc[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "43f0954f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "False"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sequences_df['ref_prot'].iloc[0] == sequences_df['alt_prot'].iloc[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "9e0fd638",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ref_prot</th>\n",
       "      <th>alt_prot</th>\n",
       "      <th>ref_dna</th>\n",
       "      <th>alt_dna</th>\n",
       "      <th>strand</th>\n",
       "      <th>label</th>\n",
       "      <th>gene</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>POS</th>\n",
       "      <th>is_ref_match</th>\n",
       "      <th>molecular_consequences</th>\n",
       "      <th>labels</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>IHSTVKPFICEVCHKSYTQFSNLCRHKRMHADCRTQIKCKDCGQMF...</td>\n",
       "      <td>IHSTVKPFICEVCHKSYTQFSNLCRHKRMHADCRTQIKCKDCGQMF...</td>\n",
       "      <td>TCCACAGCACGGTGAAGCCTTTCATATGTGAGGTCTGCCACAAGTC...</td>\n",
       "      <td>TCCACAGCACGGTGAAGCCTTTCATATGTGAGGTCTGCCACAAGTC...</td>\n",
       "      <td>+</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>PRDM16</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>3414616</td>\n",
       "      <td>1</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GNPALPLVSAVSNSSQGTTAAAGPEEKFESRLEDSCVEKLKTRSSD...</td>\n",
       "      <td>GNPALPLVSAVSNSSQGTTAAAGPEEKFESRLEDSCVEKLKTRSSD...</td>\n",
       "      <td>GGGAACCCAGCCCTGCCCCTGGTCTCCGCCGTCAGCAACAGCAGCC...</td>\n",
       "      <td>GGGAACCCAGCCCTGCCCCTGGTCTCCGCCGTCAGCAACAGCAGCC...</td>\n",
       "      <td>+</td>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>PRDM16</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>3426095</td>\n",
       "      <td>1</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GSDFEDVNTTTGTDLDTTTGTGSDLDSDVDSDPDKDKGKGKSAEGQ...</td>\n",
       "      <td>GSDFEDVNTTTGTDLDTTTGTGSDLDSDVDSDPDKDKGKGKSAEGQ...</td>\n",
       "      <td>GGCAGTGACTTTGAGGACGTCAACACCACCACGGGGACCGACCTGG...</td>\n",
       "      <td>GGCAGTGACTTTGAGGACGTCAACACCACCACGGGGACCGACCTGG...</td>\n",
       "      <td>+</td>\n",
       "      <td>Benign</td>\n",
       "      <td>PRDM16</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>3430888</td>\n",
       "      <td>1</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MEPGEVKDRILENISLSVKKLQSYFAACEDEIPAIRNHDKVLQRLC...</td>\n",
       "      <td>MEPGEVKDRILENISLSVKKLQSYFAACEDETPAIRNHDKVLQRLC...</td>\n",
       "      <td>ATGGAGCCGGGGGAGGTGAAGGACCGGATCCTGGAGAACATCTCGC...</td>\n",
       "      <td>ATGGAGCCGGGGGAGGTGAAGGACCGGATCCTGGAGAACATCTCGC...</td>\n",
       "      <td>+</td>\n",
       "      <td>Benign</td>\n",
       "      <td>PLEKHM2</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>15716271</td>\n",
       "      <td>1</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MEPGEVKDRILENISLSVKKLQSYFAACEDEIPAIRNHDKVLQRLC...</td>\n",
       "      <td>MEPGEVKDRILENISLSVKKLQSYFAACEDEIPAIRNHDKVLQRLC...</td>\n",
       "      <td>ATGGAGCCGGGGGAGGTGAAGGACCGGATCCTGGAGAACATCTCGC...</td>\n",
       "      <td>ATGGAGCCGGGGGAGGTGAAGGACCGGATCCTGGAGAACATCTCGC...</td>\n",
       "      <td>+</td>\n",
       "      <td>Benign</td>\n",
       "      <td>PLEKHM2</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>15718548</td>\n",
       "      <td>1</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1640</th>\n",
       "      <td>MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...</td>\n",
       "      <td>MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...</td>\n",
       "      <td>ATGCCTCTGCACGTGAAGTGGCCGTTCCCCGCGGTGCCGCCGCTCA...</td>\n",
       "      <td>ATGCCTCTGCACGTGAAGTGGCCGTTCCCCGCGGTGCCGCCGCTCA...</td>\n",
       "      <td>+</td>\n",
       "      <td>Pathogenic/Likely_pathogenic</td>\n",
       "      <td>TAFAZZIN</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>154413544</td>\n",
       "      <td>1</td>\n",
       "      <td>SO:0001583|missense_variant,SO:0001619|non-cod...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1641</th>\n",
       "      <td>MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...</td>\n",
       "      <td>MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...</td>\n",
       "      <td>ATGCCTCTGCACGTGAAGTGGCCGTTCCCCGCGGTGCCGCCGCTCA...</td>\n",
       "      <td>ATGCCTCTGCACGTGAAGTGGCCGTTCCCCGCGGTGCCGCCGCTCA...</td>\n",
       "      <td>+</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>TAFAZZIN</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>154413552</td>\n",
       "      <td>1</td>\n",
       "      <td>SO:0001583|missense_variant,SO:0001619|non-cod...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1642</th>\n",
       "      <td>MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...</td>\n",
       "      <td>MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...</td>\n",
       "      <td>ATGCCTCTGCACGTGAAGTGGCCGTTCCCCGCGGTGCCGCCGCTCA...</td>\n",
       "      <td>ATGCCTCTGCACGTGAAGTGGCCGTTCCCCGCGGTGCCGCCGCTCA...</td>\n",
       "      <td>+</td>\n",
       "      <td>Benign/Likely_benign</td>\n",
       "      <td>TAFAZZIN</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>154414113</td>\n",
       "      <td>0</td>\n",
       "      <td>SO:0001583|missense_variant,SO:0001627|intron_...</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1643</th>\n",
       "      <td>MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...</td>\n",
       "      <td>MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...</td>\n",
       "      <td>ATGCCTCTGCACGTGAAGTGGCCGTTCCCCGCGGTGCCGCCGCTCA...</td>\n",
       "      <td>ATGCCTCTGCACGTGAAGTGGCCGTTCCCCGCGGTGCCGCCGCTCA...</td>\n",
       "      <td>+</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>TAFAZZIN</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>154420212</td>\n",
       "      <td>1</td>\n",
       "      <td>SO:0001583|missense_variant,SO:0001619|non-cod...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1644</th>\n",
       "      <td>MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...</td>\n",
       "      <td>MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...</td>\n",
       "      <td>ATGCCTCTGCACGTGAAGTGGCCGTTCCCCGCGGTGCCGCCGCTCA...</td>\n",
       "      <td>ATGCCTCTGCACGTGAAGTGGCCGTTCCCCGCGGTGCCGCCGCTCA...</td>\n",
       "      <td>+</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>TAFAZZIN</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>154420676</td>\n",
       "      <td>1</td>\n",
       "      <td>SO:0001583|missense_variant,SO:0001619|non-cod...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1645 rows  13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                               ref_prot  \\\n",
       "0     IHSTVKPFICEVCHKSYTQFSNLCRHKRMHADCRTQIKCKDCGQMF...   \n",
       "1     GNPALPLVSAVSNSSQGTTAAAGPEEKFESRLEDSCVEKLKTRSSD...   \n",
       "2     GSDFEDVNTTTGTDLDTTTGTGSDLDSDVDSDPDKDKGKGKSAEGQ...   \n",
       "3     MEPGEVKDRILENISLSVKKLQSYFAACEDEIPAIRNHDKVLQRLC...   \n",
       "4     MEPGEVKDRILENISLSVKKLQSYFAACEDEIPAIRNHDKVLQRLC...   \n",
       "...                                                 ...   \n",
       "1640  MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...   \n",
       "1641  MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...   \n",
       "1642  MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...   \n",
       "1643  MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...   \n",
       "1644  MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...   \n",
       "\n",
       "                                               alt_prot  \\\n",
       "0     IHSTVKPFICEVCHKSYTQFSNLCRHKRMHADCRTQIKCKDCGQMF...   \n",
       "1     GNPALPLVSAVSNSSQGTTAAAGPEEKFESRLEDSCVEKLKTRSSD...   \n",
       "2     GSDFEDVNTTTGTDLDTTTGTGSDLDSDVDSDPDKDKGKGKSAEGQ...   \n",
       "3     MEPGEVKDRILENISLSVKKLQSYFAACEDETPAIRNHDKVLQRLC...   \n",
       "4     MEPGEVKDRILENISLSVKKLQSYFAACEDEIPAIRNHDKVLQRLC...   \n",
       "...                                                 ...   \n",
       "1640  MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...   \n",
       "1641  MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...   \n",
       "1642  MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...   \n",
       "1643  MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...   \n",
       "1644  MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...   \n",
       "\n",
       "                                                ref_dna  \\\n",
       "0     TCCACAGCACGGTGAAGCCTTTCATATGTGAGGTCTGCCACAAGTC...   \n",
       "1     GGGAACCCAGCCCTGCCCCTGGTCTCCGCCGTCAGCAACAGCAGCC...   \n",
       "2     GGCAGTGACTTTGAGGACGTCAACACCACCACGGGGACCGACCTGG...   \n",
       "3     ATGGAGCCGGGGGAGGTGAAGGACCGGATCCTGGAGAACATCTCGC...   \n",
       "4     ATGGAGCCGGGGGAGGTGAAGGACCGGATCCTGGAGAACATCTCGC...   \n",
       "...                                                 ...   \n",
       "1640  ATGCCTCTGCACGTGAAGTGGCCGTTCCCCGCGGTGCCGCCGCTCA...   \n",
       "1641  ATGCCTCTGCACGTGAAGTGGCCGTTCCCCGCGGTGCCGCCGCTCA...   \n",
       "1642  ATGCCTCTGCACGTGAAGTGGCCGTTCCCCGCGGTGCCGCCGCTCA...   \n",
       "1643  ATGCCTCTGCACGTGAAGTGGCCGTTCCCCGCGGTGCCGCCGCTCA...   \n",
       "1644  ATGCCTCTGCACGTGAAGTGGCCGTTCCCCGCGGTGCCGCCGCTCA...   \n",
       "\n",
       "                                                alt_dna strand  \\\n",
       "0     TCCACAGCACGGTGAAGCCTTTCATATGTGAGGTCTGCCACAAGTC...      +   \n",
       "1     GGGAACCCAGCCCTGCCCCTGGTCTCCGCCGTCAGCAACAGCAGCC...      +   \n",
       "2     GGCAGTGACTTTGAGGACGTCAACACCACCACGGGGACCGACCTGG...      +   \n",
       "3     ATGGAGCCGGGGGAGGTGAAGGACCGGATCCTGGAGAACATCTCGC...      +   \n",
       "4     ATGGAGCCGGGGGAGGTGAAGGACCGGATCCTGGAGAACATCTCGC...      +   \n",
       "...                                                 ...    ...   \n",
       "1640  ATGCCTCTGCACGTGAAGTGGCCGTTCCCCGCGGTGCCGCCGCTCA...      +   \n",
       "1641  ATGCCTCTGCACGTGAAGTGGCCGTTCCCCGCGGTGCCGCCGCTCA...      +   \n",
       "1642  ATGCCTCTGCACGTGAAGTGGCCGTTCCCCGCGGTGCCGCCGCTCA...      +   \n",
       "1643  ATGCCTCTGCACGTGAAGTGGCCGTTCCCCGCGGTGCCGCCGCTCA...      +   \n",
       "1644  ATGCCTCTGCACGTGAAGTGGCCGTTCCCCGCGGTGCCGCCGCTCA...      +   \n",
       "\n",
       "                             label      gene ref alt        POS  is_ref_match  \\\n",
       "0                       Pathogenic    PRDM16   T   C    3414616             1   \n",
       "1                    Likely_benign    PRDM16   G   A    3426095             1   \n",
       "2                           Benign    PRDM16   G   A    3430888             1   \n",
       "3                           Benign   PLEKHM2   T   C   15716271             1   \n",
       "4                           Benign   PLEKHM2   G   A   15718548             1   \n",
       "...                            ...       ...  ..  ..        ...           ...   \n",
       "1640  Pathogenic/Likely_pathogenic  TAFAZZIN   G   A  154413544             1   \n",
       "1641             Likely_pathogenic  TAFAZZIN   G   A  154413552             1   \n",
       "1642          Benign/Likely_benign  TAFAZZIN   T   C  154414113             0   \n",
       "1643             Likely_pathogenic  TAFAZZIN   G   T  154420212             1   \n",
       "1644                    Pathogenic  TAFAZZIN   G   C  154420676             1   \n",
       "\n",
       "                                 molecular_consequences  labels  \n",
       "0                           SO:0001583|missense_variant       1  \n",
       "1                           SO:0001583|missense_variant       0  \n",
       "2                           SO:0001583|missense_variant       0  \n",
       "3                           SO:0001583|missense_variant       0  \n",
       "4                           SO:0001583|missense_variant       0  \n",
       "...                                                 ...     ...  \n",
       "1640  SO:0001583|missense_variant,SO:0001619|non-cod...       1  \n",
       "1641  SO:0001583|missense_variant,SO:0001619|non-cod...       1  \n",
       "1642  SO:0001583|missense_variant,SO:0001627|intron_...       0  \n",
       "1643  SO:0001583|missense_variant,SO:0001619|non-cod...       1  \n",
       "1644  SO:0001583|missense_variant,SO:0001619|non-cod...       1  \n",
       "\n",
       "[1645 rows x 13 columns]"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sequences_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "98c96b35",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                               ref_prot  \\\n",
      "0     IHSTVKPFICEVCHKSYTQFSNLCRHKRMHADCRTQIKCKDCGQMF...   \n",
      "1     GNPALPLVSAVSNSSQGTTAAAGPEEKFESRLEDSCVEKLKTRSSD...   \n",
      "2     GSDFEDVNTTTGTDLDTTTGTGSDLDSDVDSDPDKDKGKGKSAEGQ...   \n",
      "3     MEPGEVKDRILENISLSVKKLQSYFAACEDEIPAIRNHDKVLQRLC...   \n",
      "4     MEPGEVKDRILENISLSVKKLQSYFAACEDEIPAIRNHDKVLQRLC...   \n",
      "...                                                 ...   \n",
      "1640  MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...   \n",
      "1641  MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...   \n",
      "1642  MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...   \n",
      "1643  MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...   \n",
      "1644  MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...   \n",
      "\n",
      "                                               alt_prot  \\\n",
      "0     IHSTVKPFICEVCHKSYTQFSNLCRHKRMHADCRTQIKCKDCGQMF...   \n",
      "1     GNPALPLVSAVSNSSQGTTAAAGPEEKFESRLEDSCVEKLKTRSSD...   \n",
      "2     GSDFEDVNTTTGTDLDTTTGTGSDLDSDVDSDPDKDKGKGKSAEGQ...   \n",
      "3     MEPGEVKDRILENISLSVKKLQSYFAACEDETPAIRNHDKVLQRLC...   \n",
      "4     MEPGEVKDRILENISLSVKKLQSYFAACEDEIPAIRNHDKVLQRLC...   \n",
      "...                                                 ...   \n",
      "1640  MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...   \n",
      "1641  MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...   \n",
      "1642  MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...   \n",
      "1643  MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...   \n",
      "1644  MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...   \n",
      "\n",
      "                                                ref_dna  \\\n",
      "0     TCCACAGCACGGTGAAGCCTTTCATATGTGAGGTCTGCCACAAGTC...   \n",
      "1     GGGAACCCAGCCCTGCCCCTGGTCTCCGCCGTCAGCAACAGCAGCC...   \n",
      "2     GGCAGTGACTTTGAGGACGTCAACACCACCACGGGGACCGACCTGG...   \n",
      "3     ATGGAGCCGGGGGAGGTGAAGGACCGGATCCTGGAGAACATCTCGC...   \n",
      "4     ATGGAGCCGGGGGAGGTGAAGGACCGGATCCTGGAGAACATCTCGC...   \n",
      "...                                                 ...   \n",
      "1640  ATGCCTCTGCACGTGAAGTGGCCGTTCCCCGCGGTGCCGCCGCTCA...   \n",
      "1641  ATGCCTCTGCACGTGAAGTGGCCGTTCCCCGCGGTGCCGCCGCTCA...   \n",
      "1642  ATGCCTCTGCACGTGAAGTGGCCGTTCCCCGCGGTGCCGCCGCTCA...   \n",
      "1643  ATGCCTCTGCACGTGAAGTGGCCGTTCCCCGCGGTGCCGCCGCTCA...   \n",
      "1644  ATGCCTCTGCACGTGAAGTGGCCGTTCCCCGCGGTGCCGCCGCTCA...   \n",
      "\n",
      "                                                alt_dna strand  \\\n",
      "0     TCCACAGCACGGTGAAGCCTTTCATATGTGAGGTCTGCCACAAGTC...      +   \n",
      "1     GGGAACCCAGCCCTGCCCCTGGTCTCCGCCGTCAGCAACAGCAGCC...      +   \n",
      "2     GGCAGTGACTTTGAGGACGTCAACACCACCACGGGGACCGACCTGG...      +   \n",
      "3     ATGGAGCCGGGGGAGGTGAAGGACCGGATCCTGGAGAACATCTCGC...      +   \n",
      "4     ATGGAGCCGGGGGAGGTGAAGGACCGGATCCTGGAGAACATCTCGC...      +   \n",
      "...                                                 ...    ...   \n",
      "1640  ATGCCTCTGCACGTGAAGTGGCCGTTCCCCGCGGTGCCGCCGCTCA...      +   \n",
      "1641  ATGCCTCTGCACGTGAAGTGGCCGTTCCCCGCGGTGCCGCCGCTCA...      +   \n",
      "1642  ATGCCTCTGCACGTGAAGTGGCCGTTCCCCGCGGTGCCGCCGCTCA...      +   \n",
      "1643  ATGCCTCTGCACGTGAAGTGGCCGTTCCCCGCGGTGCCGCCGCTCA...      +   \n",
      "1644  ATGCCTCTGCACGTGAAGTGGCCGTTCCCCGCGGTGCCGCCGCTCA...      +   \n",
      "\n",
      "                             label      gene ref alt        POS  is_ref_match  \\\n",
      "0                       Pathogenic    PRDM16   T   C    3414616             1   \n",
      "1                    Likely_benign    PRDM16   G   A    3426095             1   \n",
      "2                           Benign    PRDM16   G   A    3430888             1   \n",
      "3                           Benign   PLEKHM2   T   C   15716271             1   \n",
      "4                           Benign   PLEKHM2   G   A   15718548             1   \n",
      "...                            ...       ...  ..  ..        ...           ...   \n",
      "1640  Pathogenic/Likely_pathogenic  TAFAZZIN   G   A  154413544             1   \n",
      "1641             Likely_pathogenic  TAFAZZIN   G   A  154413552             1   \n",
      "1642          Benign/Likely_benign  TAFAZZIN   T   C  154414113             0   \n",
      "1643             Likely_pathogenic  TAFAZZIN   G   T  154420212             1   \n",
      "1644                    Pathogenic  TAFAZZIN   G   C  154420676             1   \n",
      "\n",
      "                                 molecular_consequences  labels  \n",
      "0                           SO:0001583|missense_variant       1  \n",
      "1                           SO:0001583|missense_variant       0  \n",
      "2                           SO:0001583|missense_variant       0  \n",
      "3                           SO:0001583|missense_variant       0  \n",
      "4                           SO:0001583|missense_variant       0  \n",
      "...                                                 ...     ...  \n",
      "1640  SO:0001583|missense_variant,SO:0001619|non-cod...       1  \n",
      "1641  SO:0001583|missense_variant,SO:0001619|non-cod...       1  \n",
      "1642  SO:0001583|missense_variant,SO:0001627|intron_...       0  \n",
      "1643  SO:0001583|missense_variant,SO:0001619|non-cod...       1  \n",
      "1644  SO:0001583|missense_variant,SO:0001619|non-cod...       1  \n",
      "\n",
      "[1645 rows x 13 columns]\n"
     ]
    }
   ],
   "source": [
    "# Adjust the mapping to account for variations in the 'cln_significance' descriptions\n",
    "def map_significance(value):\n",
    "    value = value.lower()  # Convert to lower case to handle case variations uniformly\n",
    "    if 'benign' in value:  # This will match \"Benign\" and \"Likely_benign\"\n",
    "        return 0\n",
    "    else:  # Matches \"Pathogenic\"\n",
    "        return 1\n",
    "#sequences_df_rc['seq_a'] = sequences_df_rc['seq_a'].str.upper()\n",
    "#sequences_df_rc['seq_b'] = sequences_df_rc['seq_b'].str.upper()\n",
    "# Apply the mapping function to the 'cln_significance' column\n",
    "sequences_df['labels'] = sequences_df['label'].apply(map_significance)\n",
    "#sequences_df_rc['POS'] = sequences_df['POS']\n",
    "#sequences_df_reset = sequences_df_rc.reset_index(drop=True)\n",
    "print(sequences_df)\n",
    "# Now sequences_df_reset contains the appended 'cln_significance_mapped' column\n",
    "sequences_df_rc = sequences_df[['ref_prot','alt_prot','labels']]\n",
    "sequences_df_rc_gene = sequences_df[['ref_prot','alt_prot','labels','gene','label']]\n",
    "sequences_df_rc = sequences_df_rc.rename(columns={'ref_prot': 'wt_seq', 'alt_prot': 'mut_seq'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "3d99db9b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>wt_seq</th>\n",
       "      <th>mut_seq</th>\n",
       "      <th>labels</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>IHSTVKPFICEVCHKSYTQFSNLCRHKRMHADCRTQIKCKDCGQMF...</td>\n",
       "      <td>IHSTVKPFICEVCHKSYTQFSNLCRHKRMHADCRTQIKCKDCGQMF...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GNPALPLVSAVSNSSQGTTAAAGPEEKFESRLEDSCVEKLKTRSSD...</td>\n",
       "      <td>GNPALPLVSAVSNSSQGTTAAAGPEEKFESRLEDSCVEKLKTRSSD...</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GSDFEDVNTTTGTDLDTTTGTGSDLDSDVDSDPDKDKGKGKSAEGQ...</td>\n",
       "      <td>GSDFEDVNTTTGTDLDTTTGTGSDLDSDVDSDPDKDKGKGKSAEGQ...</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MEPGEVKDRILENISLSVKKLQSYFAACEDEIPAIRNHDKVLQRLC...</td>\n",
       "      <td>MEPGEVKDRILENISLSVKKLQSYFAACEDETPAIRNHDKVLQRLC...</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MEPGEVKDRILENISLSVKKLQSYFAACEDEIPAIRNHDKVLQRLC...</td>\n",
       "      <td>MEPGEVKDRILENISLSVKKLQSYFAACEDEIPAIRNHDKVLQRLC...</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1640</th>\n",
       "      <td>MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...</td>\n",
       "      <td>MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1641</th>\n",
       "      <td>MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...</td>\n",
       "      <td>MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1642</th>\n",
       "      <td>MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...</td>\n",
       "      <td>MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1643</th>\n",
       "      <td>MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...</td>\n",
       "      <td>MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1644</th>\n",
       "      <td>MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...</td>\n",
       "      <td>MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1635 rows  3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                 wt_seq  \\\n",
       "0     IHSTVKPFICEVCHKSYTQFSNLCRHKRMHADCRTQIKCKDCGQMF...   \n",
       "1     GNPALPLVSAVSNSSQGTTAAAGPEEKFESRLEDSCVEKLKTRSSD...   \n",
       "2     GSDFEDVNTTTGTDLDTTTGTGSDLDSDVDSDPDKDKGKGKSAEGQ...   \n",
       "3     MEPGEVKDRILENISLSVKKLQSYFAACEDEIPAIRNHDKVLQRLC...   \n",
       "4     MEPGEVKDRILENISLSVKKLQSYFAACEDEIPAIRNHDKVLQRLC...   \n",
       "...                                                 ...   \n",
       "1640  MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...   \n",
       "1641  MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...   \n",
       "1642  MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...   \n",
       "1643  MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...   \n",
       "1644  MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...   \n",
       "\n",
       "                                                mut_seq  labels  \n",
       "0     IHSTVKPFICEVCHKSYTQFSNLCRHKRMHADCRTQIKCKDCGQMF...       1  \n",
       "1     GNPALPLVSAVSNSSQGTTAAAGPEEKFESRLEDSCVEKLKTRSSD...       0  \n",
       "2     GSDFEDVNTTTGTDLDTTTGTGSDLDSDVDSDPDKDKGKGKSAEGQ...       0  \n",
       "3     MEPGEVKDRILENISLSVKKLQSYFAACEDETPAIRNHDKVLQRLC...       0  \n",
       "4     MEPGEVKDRILENISLSVKKLQSYFAACEDEIPAIRNHDKVLQRLC...       0  \n",
       "...                                                 ...     ...  \n",
       "1640  MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...       1  \n",
       "1641  MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...       1  \n",
       "1642  MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...       0  \n",
       "1643  MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...       1  \n",
       "1644  MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...       1  \n",
       "\n",
       "[1635 rows x 3 columns]"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sequences_df_rc.to_csv('/common/zhangz2lab/zhanh/esm-variants/cropped/Clinvar_CM_protein.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "d3849404",
   "metadata": {},
   "outputs": [],
   "source": [
    "sequences_df_rc = sequences_df_rc[sequences_df_rc['mut_seq'] != '']\n",
    "sequences_df_rc = sequences_df_rc[sequences_df_rc['wt_seq'] != '']\n",
    "sequences_df_rc_gene = sequences_df_rc_gene[sequences_df_rc_gene['ref_prot'] != '']\n",
    "sequences_df_rc_gene = sequences_df_rc_gene[sequences_df_rc_gene['alt_prot'] != '']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "e6398ffc",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>wt_seq</th>\n",
       "      <th>mut_seq</th>\n",
       "      <th>labels</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>IHSTVKPFICEVCHKSYTQFSNLCRHKRMHADCRTQIKCKDCGQMF...</td>\n",
       "      <td>IHSTVKPFICEVCHKSYTQFSNLCRHKRMHADCRTQIKCKDCGQMF...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GNPALPLVSAVSNSSQGTTAAAGPEEKFESRLEDSCVEKLKTRSSD...</td>\n",
       "      <td>GNPALPLVSAVSNSSQGTTAAAGPEEKFESRLEDSCVEKLKTRSSD...</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GSDFEDVNTTTGTDLDTTTGTGSDLDSDVDSDPDKDKGKGKSAEGQ...</td>\n",
       "      <td>GSDFEDVNTTTGTDLDTTTGTGSDLDSDVDSDPDKDKGKGKSAEGQ...</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MEPGEVKDRILENISLSVKKLQSYFAACEDEIPAIRNHDKVLQRLC...</td>\n",
       "      <td>MEPGEVKDRILENISLSVKKLQSYFAACEDETPAIRNHDKVLQRLC...</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MEPGEVKDRILENISLSVKKLQSYFAACEDEIPAIRNHDKVLQRLC...</td>\n",
       "      <td>MEPGEVKDRILENISLSVKKLQSYFAACEDEIPAIRNHDKVLQRLC...</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1630</th>\n",
       "      <td>MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...</td>\n",
       "      <td>MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1631</th>\n",
       "      <td>MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...</td>\n",
       "      <td>MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1632</th>\n",
       "      <td>MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...</td>\n",
       "      <td>MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1633</th>\n",
       "      <td>MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...</td>\n",
       "      <td>MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1634</th>\n",
       "      <td>MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...</td>\n",
       "      <td>MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1635 rows  3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                 wt_seq  \\\n",
       "0     IHSTVKPFICEVCHKSYTQFSNLCRHKRMHADCRTQIKCKDCGQMF...   \n",
       "1     GNPALPLVSAVSNSSQGTTAAAGPEEKFESRLEDSCVEKLKTRSSD...   \n",
       "2     GSDFEDVNTTTGTDLDTTTGTGSDLDSDVDSDPDKDKGKGKSAEGQ...   \n",
       "3     MEPGEVKDRILENISLSVKKLQSYFAACEDEIPAIRNHDKVLQRLC...   \n",
       "4     MEPGEVKDRILENISLSVKKLQSYFAACEDEIPAIRNHDKVLQRLC...   \n",
       "...                                                 ...   \n",
       "1630  MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...   \n",
       "1631  MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...   \n",
       "1632  MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...   \n",
       "1633  MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...   \n",
       "1634  MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...   \n",
       "\n",
       "                                                mut_seq  labels  \n",
       "0     IHSTVKPFICEVCHKSYTQFSNLCRHKRMHADCRTQIKCKDCGQMF...       1  \n",
       "1     GNPALPLVSAVSNSSQGTTAAAGPEEKFESRLEDSCVEKLKTRSSD...       0  \n",
       "2     GSDFEDVNTTTGTDLDTTTGTGSDLDSDVDSDPDKDKGKGKSAEGQ...       0  \n",
       "3     MEPGEVKDRILENISLSVKKLQSYFAACEDETPAIRNHDKVLQRLC...       0  \n",
       "4     MEPGEVKDRILENISLSVKKLQSYFAACEDEIPAIRNHDKVLQRLC...       0  \n",
       "...                                                 ...     ...  \n",
       "1630  MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...       1  \n",
       "1631  MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...       1  \n",
       "1632  MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...       0  \n",
       "1633  MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...       1  \n",
       "1634  MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...       1  \n",
       "\n",
       "[1635 rows x 3 columns]"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sequences_df_reset = sequences_df_rc.reset_index(drop=True)\n",
    "sequences_df_reset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "f0559438",
   "metadata": {},
   "outputs": [],
   "source": [
    "sequences_df_rc.to_csv('/common/zhangz2lab/zhanh/esm-variants/cropped/Clinvar_CM_protein.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "bb76b432",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ref_prot</th>\n",
       "      <th>alt_prot</th>\n",
       "      <th>labels</th>\n",
       "      <th>gene</th>\n",
       "      <th>label</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>IHSTVKPFICEVCHKSYTQFSNLCRHKRMHADCRTQIKCKDCGQMF...</td>\n",
       "      <td>IHSTVKPFICEVCHKSYTQFSNLCRHKRMHADCRTQIKCKDCGQMF...</td>\n",
       "      <td>1</td>\n",
       "      <td>PRDM16</td>\n",
       "      <td>Pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GNPALPLVSAVSNSSQGTTAAAGPEEKFESRLEDSCVEKLKTRSSD...</td>\n",
       "      <td>GNPALPLVSAVSNSSQGTTAAAGPEEKFESRLEDSCVEKLKTRSSD...</td>\n",
       "      <td>0</td>\n",
       "      <td>PRDM16</td>\n",
       "      <td>Likely_benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GSDFEDVNTTTGTDLDTTTGTGSDLDSDVDSDPDKDKGKGKSAEGQ...</td>\n",
       "      <td>GSDFEDVNTTTGTDLDTTTGTGSDLDSDVDSDPDKDKGKGKSAEGQ...</td>\n",
       "      <td>0</td>\n",
       "      <td>PRDM16</td>\n",
       "      <td>Benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MEPGEVKDRILENISLSVKKLQSYFAACEDEIPAIRNHDKVLQRLC...</td>\n",
       "      <td>MEPGEVKDRILENISLSVKKLQSYFAACEDETPAIRNHDKVLQRLC...</td>\n",
       "      <td>0</td>\n",
       "      <td>PLEKHM2</td>\n",
       "      <td>Benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MEPGEVKDRILENISLSVKKLQSYFAACEDEIPAIRNHDKVLQRLC...</td>\n",
       "      <td>MEPGEVKDRILENISLSVKKLQSYFAACEDEIPAIRNHDKVLQRLC...</td>\n",
       "      <td>0</td>\n",
       "      <td>PLEKHM2</td>\n",
       "      <td>Benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1640</th>\n",
       "      <td>MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...</td>\n",
       "      <td>MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...</td>\n",
       "      <td>1</td>\n",
       "      <td>TAFAZZIN</td>\n",
       "      <td>Pathogenic/Likely_pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1641</th>\n",
       "      <td>MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...</td>\n",
       "      <td>MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...</td>\n",
       "      <td>1</td>\n",
       "      <td>TAFAZZIN</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1642</th>\n",
       "      <td>MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...</td>\n",
       "      <td>MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...</td>\n",
       "      <td>0</td>\n",
       "      <td>TAFAZZIN</td>\n",
       "      <td>Benign/Likely_benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1643</th>\n",
       "      <td>MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...</td>\n",
       "      <td>MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...</td>\n",
       "      <td>1</td>\n",
       "      <td>TAFAZZIN</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1644</th>\n",
       "      <td>MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...</td>\n",
       "      <td>MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...</td>\n",
       "      <td>1</td>\n",
       "      <td>TAFAZZIN</td>\n",
       "      <td>Pathogenic</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1635 rows  5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                               ref_prot  \\\n",
       "0     IHSTVKPFICEVCHKSYTQFSNLCRHKRMHADCRTQIKCKDCGQMF...   \n",
       "1     GNPALPLVSAVSNSSQGTTAAAGPEEKFESRLEDSCVEKLKTRSSD...   \n",
       "2     GSDFEDVNTTTGTDLDTTTGTGSDLDSDVDSDPDKDKGKGKSAEGQ...   \n",
       "3     MEPGEVKDRILENISLSVKKLQSYFAACEDEIPAIRNHDKVLQRLC...   \n",
       "4     MEPGEVKDRILENISLSVKKLQSYFAACEDEIPAIRNHDKVLQRLC...   \n",
       "...                                                 ...   \n",
       "1640  MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...   \n",
       "1641  MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...   \n",
       "1642  MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...   \n",
       "1643  MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...   \n",
       "1644  MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...   \n",
       "\n",
       "                                               alt_prot  labels      gene  \\\n",
       "0     IHSTVKPFICEVCHKSYTQFSNLCRHKRMHADCRTQIKCKDCGQMF...       1    PRDM16   \n",
       "1     GNPALPLVSAVSNSSQGTTAAAGPEEKFESRLEDSCVEKLKTRSSD...       0    PRDM16   \n",
       "2     GSDFEDVNTTTGTDLDTTTGTGSDLDSDVDSDPDKDKGKGKSAEGQ...       0    PRDM16   \n",
       "3     MEPGEVKDRILENISLSVKKLQSYFAACEDETPAIRNHDKVLQRLC...       0   PLEKHM2   \n",
       "4     MEPGEVKDRILENISLSVKKLQSYFAACEDEIPAIRNHDKVLQRLC...       0   PLEKHM2   \n",
       "...                                                 ...     ...       ...   \n",
       "1640  MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...       1  TAFAZZIN   \n",
       "1641  MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...       1  TAFAZZIN   \n",
       "1642  MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...       0  TAFAZZIN   \n",
       "1643  MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...       1  TAFAZZIN   \n",
       "1644  MPLHVKWPFPAVPPLTWTLASSVVMGLVGTYSCFWTKYMNHLTVHN...       1  TAFAZZIN   \n",
       "\n",
       "                             label  \n",
       "0                       Pathogenic  \n",
       "1                    Likely_benign  \n",
       "2                           Benign  \n",
       "3                           Benign  \n",
       "4                           Benign  \n",
       "...                            ...  \n",
       "1640  Pathogenic/Likely_pathogenic  \n",
       "1641             Likely_pathogenic  \n",
       "1642          Benign/Likely_benign  \n",
       "1643             Likely_pathogenic  \n",
       "1644                    Pathogenic  \n",
       "\n",
       "[1635 rows x 5 columns]"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sequences_df_rc_gene"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "id": "5cb57d6b",
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv(\"/common/zhangz2lab/zhanh/Propath-2/PLLR_CM.csv\")\n",
    "# For the DataFrame loaded from CSV\n",
    "\n",
    "# For the sequences_df DataFrame\n",
    "sequences_df_rc_gene.reset_index(drop=True, inplace=True)\n",
    "df['gene'] = sequences_df_rc_gene['gene']\n",
    "df['label'] = sequences_df_rc_gene['label']\n",
    "df.to_csv('/common/zhangz2lab/zhanh/Propath-2/PLLR_CM.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "b8b7dfda",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0    1113\n",
      "1     522\n",
      "Name: true_labels_callback, dtype: int64\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "df = pd.read_csv('/common/zhangz2lab/zhanh/Propath-2/PLLR_CM.csv')\n",
    "value_counts = df['true_labels_callback'].value_counts()\n",
    "\n",
    "# Print the counts\n",
    "print(value_counts)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "id": "6b0d9944",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0         PRDM16\n",
       "1         PRDM16\n",
       "2         PRDM16\n",
       "3        PLEKHM2\n",
       "4        PLEKHM2\n",
       "          ...   \n",
       "1630    TAFAZZIN\n",
       "1631    TAFAZZIN\n",
       "1632    TAFAZZIN\n",
       "1633    TAFAZZIN\n",
       "1634    TAFAZZIN\n",
       "Name: gene, Length: 1635, dtype: object"
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "29781bea",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
